{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d68a2faa",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Open parser.py file and change the code for from parser import Parser to from .parser import Parser\n",
    "# .conda\\envs\\untl_py3\\Lib\\site-packages\\vcf\n",
    "\n",
    "#import vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8c834b4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#\"S:\\\\UNTLSynology\\Quant\\Data\\gnomAd\"\n",
    "\n",
    "#vcf.Reader(open(\":\\\\UNTLSynology\\Quant\\Data\\gnomAd\", 'r'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f260789b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_variants_by_genes(vcf_file, output_file, genes_of_interest):\n",
    "    \"\"\"\n",
    "    Extract variants from a VCF file based on specified gene names.\n",
    "    \n",
    "    Parameters:\n",
    "    - vcf_file: Input VCF file path.\n",
    "    - output_file: Output VCF file path.\n",
    "    - genes_of_interest: List of gene names you want to extract variants for.\n",
    "    \"\"\"\n",
    "    with open(output_file, 'w') as out_vcf:\n",
    "        vcf_reader = vcf.Reader(open(vcf_file, 'r'))\n",
    "        \n",
    "        # Write VCF header to the output file\n",
    "        vcf.Writer(out_vcf, vcf_reader)\n",
    "        \n",
    "        # Iterate through each variant in the input VCF file\n",
    "        for record in vcf_reader:\n",
    "            # Check if the variant has gene information in the INFO field\n",
    "            if 'GENE' in record.INFO:\n",
    "                # Extract the gene names for the variant\n",
    "                gene_names = record.INFO['GENE'].split(',')\n",
    "                \n",
    "                # Check if any of the gene names match the genes of interest\n",
    "                if any(gene in genes_of_interest for gene in gene_names):\n",
    "                    # Write the variant to the output VCF file\n",
    "                    out_vcf.write_record(record)\n",
    "\n",
    "# Example usage\n",
    "input_vcf_file = 'your_input.vcf'\n",
    "output_vcf_file = 'output_filtered.vcf'\n",
    "genes_of_interest = ['GeneA', 'GeneB']\n",
    "\n",
    "#extract_variants_by_genes(input_vcf_file, output_vcf_file, genes_of_interest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d9392854",
   "metadata": {},
   "outputs": [],
   "source": [
    "import gzip\n",
    "import pandas as pd\n",
    "\n",
    "row = \"\"\n",
    "\n",
    "def extract_variants_by_gene(vcf_bgz_file, gene_name):\n",
    "    \"\"\"\n",
    "    Extract variants from a bgzipped VCF file based on a specified gene name.\n",
    "    \n",
    "    Parameters:\n",
    "    - vcf_bgz_file: Input bgzipped VCF file path.\n",
    "    - gene_name: Gene name to filter variants.\n",
    "    \n",
    "    Returns:\n",
    "    - Pandas DataFrame containing variants for the specified gene.\n",
    "    \"\"\"\n",
    "    global row\n",
    "    variants = []\n",
    "    i = 0\n",
    "    with gzip.open(vcf_bgz_file, 'rt') as vcf_file:\n",
    "        for line in vcf_file:\n",
    "            if line.startswith('#'):\n",
    "                continue\n",
    "            row = line\n",
    "            i += 1\n",
    "            #if i > 100: break\n",
    "            fields = line.strip().split('\\t')\n",
    "            if (int(fields[1]) >= 44943683) and (int(fields[1]) <= 44943837):\n",
    "                \n",
    "                print(fields[5])\n",
    "                \n",
    "                break\n",
    "            # Extract gene names from the INFO field\n",
    "            info_field = fields[7]\n",
    "            continue\n",
    "            if gene_name in gene_names:\n",
    "                variants.append({\n",
    "                    'Chromosome': fields[0],\n",
    "                    'Position': int(fields[1]),\n",
    "                    'rsIDs': fields[2],\n",
    "                    'ALT': fields[4],\n",
    "                    'QUAL': fields[5],\n",
    "                    'FILTER': fields[6],\n",
    "                    'INFO': info_field\n",
    "                })\n",
    "    return pd.DataFrame(variants)\n",
    "\n",
    "def save_variants_to_file(variants_df, output_file):\n",
    "    \"\"\"\n",
    "    Save variants DataFrame to a CSV or Excel file.\n",
    "    \n",
    "    Parameters:\n",
    "    - variants_df: Pandas DataFrame containing variants.\n",
    "    - output_file: Output file path.\n",
    "    \"\"\"\n",
    "    if output_file.endswith('.csv'):\n",
    "        variants_df.to_csv(output_file, index=False)\n",
    "    elif output_file.endswith('.xlsx'):\n",
    "        variants_df.to_excel(output_file, index=False)\n",
    "    else:\n",
    "        raise ValueError(\"Output file format not supported. Please use either CSV or Excel.\")\n",
    "\n",
    "# Example usage\n",
    "\n",
    "input_bgz_vcf_file = 'C:\\\\Users\\\\Administrator\\\\Downloads\\\\gnomad.genomes.v4.0.sites.chr21.vcf.bgz'\n",
    "gene_name = 'GeneA'\n",
    "output_file = 'variants_for_geneA.xlsx'\n",
    "\n",
    "#variants_df = extract_variants_by_gene(input_bgz_vcf_file, gene_name)\n",
    "#save_variants_to_file(variants_df, output_file)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1fa9be0d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08347ac1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1e0b7bfe",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b4ac76a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "start first genename CH507-9B2.3,CH507-9B2.4 5124755 5124875\n",
      "Find one matching CH507-9B2.3,CH507-9B2.4 5124755 5124875 current position 5124863\n",
      "No matching anymore CH507-9B2.3,CH507-9B2.4 5124755 5124875 current position 5125038\n",
      "start next genename LIPI 14163418 14163523\n",
      "Find one matching LIPI 14163418 14163523 current position 14163418\n",
      "Find one matching LIPI 14163418 14163523 current position 14163430\n",
      "Find one matching LIPI 14163418 14163523 current position 14163435\n",
      "Find one matching LIPI 14163418 14163523 current position 14163436\n",
      "Find one matching LIPI 14163418 14163523 current position 14163440\n",
      "Find one matching LIPI 14163418 14163523 current position 14163448\n",
      "Find one matching LIPI 14163418 14163523 current position 14163453\n",
      "Find one matching LIPI 14163418 14163523 current position 14163454\n",
      "Find one matching LIPI 14163418 14163523 current position 14163455\n",
      "Find one matching LIPI 14163418 14163523 current position 14163457\n",
      "Find one matching LIPI 14163418 14163523 current position 14163468\n",
      "Find one matching LIPI 14163418 14163523 current position 14163469\n",
      "Find one matching LIPI 14163418 14163523 current position 14163471\n",
      "Find one matching LIPI 14163418 14163523 current position 14163476\n",
      "Find one matching LIPI 14163418 14163523 current position 14163479\n",
      "Find one matching LIPI 14163418 14163523 current position 14163480\n",
      "Find one matching LIPI 14163418 14163523 current position 14163487\n",
      "Find one matching LIPI 14163418 14163523 current position 14163497\n",
      "Find one matching LIPI 14163418 14163523 current position 14163510\n",
      "Find one matching LIPI 14163418 14163523 current position 14163515\n",
      "Find one matching LIPI 14163418 14163523 current position 14163516\n",
      "No matching anymore LIPI 14163418 14163523 current position 14163531\n",
      "start next genename JAM2 25702169 25702269\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "vcf_bgz_file = 'C:\\\\Users\\\\Administrator\\\\Downloads\\\\gnomad.genomes.v4.0.sites.chr21.vcf.bgz'\n",
    "\n",
    "gtex_df = pd.read_csv(\"ALTssDB_5ss.csv\")\n",
    "\n",
    "gtex_df.head(10)\n",
    "\n",
    "gtex_df = gtex_df[gtex_df['chromo'] == 'chr21'].copy()\n",
    "\n",
    "gtex_df = gtex_df.sort_values(\"start\")\n",
    "\n",
    "# List to store merged rows\n",
    "merged_rows = []\n",
    "\n",
    "i = 0\n",
    "\n",
    "row1 = gtex_df.iloc[i]\n",
    "\n",
    "genename = row1['genename']\n",
    "       \n",
    "start, end = row1['start'], row1['end']\n",
    "\n",
    "#start, end = 44943683, 44943837\n",
    "\n",
    "    \n",
    "with gzip.open(vcf_bgz_file, 'rt') as vcf_file:\n",
    "    \n",
    "    print(\"start first genename\", genename, start, end)\n",
    "            \n",
    "    for line in vcf_file:\n",
    "\n",
    "        if line.startswith('#'):\n",
    "            continue\n",
    "\n",
    "        a = line.find(\"\\t\", 0 + 1)\n",
    "\n",
    "        b = line.find(\"\\t\", a + 1)\n",
    "\n",
    "        if a >= b: continue\n",
    "\n",
    "        Chromosome, Position = line[:a], line[a+1: b]\n",
    "        \n",
    "        if int(Position) > gtex_df.iloc[len(gtex_df)-1]['end']:\n",
    "            \n",
    "            print(\"Finish the searching\")\n",
    "            \n",
    "            break\n",
    "\n",
    "        if int(Position) > end:\n",
    "\n",
    "            print(\"No matching anymore\", genename, start, end, \"current position\", Position)\n",
    "            \n",
    "            i = i + 1\n",
    "            \n",
    "            row1 = gtex_df.iloc[i]\n",
    "\n",
    "            genename = row1['genename']\n",
    "\n",
    "            start, end = row1['start'], row1['end']\n",
    "\n",
    "            print(\"start next genename\", genename, start, end)\n",
    "\n",
    "\n",
    "        if (int(Position) >= start) and (int(Position) <= end):\n",
    "\n",
    "            print(\"Find one matching\", genename, start, end, \"current position\", Position)\n",
    "\n",
    "            combined_row = row1.to_dict()\n",
    "\n",
    "            fields = line.strip().split('\\t')\n",
    "\n",
    "\n",
    "            combined_row.update({\"Chromosome\": fields[0]})\n",
    "\n",
    "            combined_row.update({\"Position\": fields[1]})\n",
    "\n",
    "            combined_row.update({\"rsIDs\": fields[2]})\n",
    "\n",
    "            combined_row.update({\"Reference\": fields[3]})\n",
    "\n",
    "            combined_row.update({\"Alternate\": fields[4]})\n",
    "\n",
    "            last_field = fields[-1]\n",
    "\n",
    "            a = last_field.find(\"AC=\")\n",
    "\n",
    "            a += len(\"AC=\")\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Count\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AN=\")\n",
    "\n",
    "            a += len(\"AN=\")\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Number\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AF=\")\n",
    "\n",
    "            a += len(\"AF=\")\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Frequency\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "\n",
    "            a = last_field.find(\"AC_ami=\")\n",
    "\n",
    "            a += (len(\"AC_ami=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Count Amish\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AN_ami=\")\n",
    "\n",
    "            a += (len(\"AN_ami=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Number Amish\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AF_ami=\")\n",
    "\n",
    "            a += (len(\"AF_ami=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Frequency Amish\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "            a = last_field.find(\"AC_amr=\")\n",
    "\n",
    "            a += (len(\"AC_amr=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Count Admixed American\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AN_amr=\")\n",
    "\n",
    "            a += (len(\"AN_amr=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Number Admixed American\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AF_amr=\")\n",
    "\n",
    "            a += (len(\"AF_amr=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Frequency Admixed American\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "            a = last_field.find(\"AC_asj=\")\n",
    "\n",
    "            a += (len(\"AC_asj=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Count Ashkenazi Jewish\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AN_asj=\")\n",
    "\n",
    "            a += (len(\"AN_asj=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Number Ashkenazi Jewish\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AF_asj=\")\n",
    "\n",
    "            a += (len(\"AF_asj=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Frequency Ashkenazi Jewish\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "\n",
    "            a = last_field.find(\"AC_eas=\")\n",
    "\n",
    "            a += (len(\"AC_eas=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Count East Asian\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AN_eas=\")\n",
    "\n",
    "            a += (len(\"AN_eas=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Number East Asian\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AF_eas=\")\n",
    "\n",
    "            a += (len(\"AF_eas=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Frequency East Asian\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "            a = last_field.find(\"AC_fin=\")\n",
    "\n",
    "            a += (len(\"AC_fin=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Count European (Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AN_fin=\")\n",
    "\n",
    "            a += (len(\"AN_fin=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Number European (Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AF_fin=\")\n",
    "\n",
    "            a += (len(\"AF_fin=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Frequency European (Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "            a = last_field.find(\"AC_nfe=\")\n",
    "\n",
    "            a += (len(\"AC_nfe=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Count European (non-Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AN_nfe=\")\n",
    "\n",
    "            a += (len(\"AN_nfe=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Number European (non-Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AF_nfe=\")\n",
    "\n",
    "            a += (len(\"AF_nfe=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Frequency European (non-Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "            a = last_field.find(\"AC_sas=\")\n",
    "\n",
    "            a += (len(\"AC_sas=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Count South Asian\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AN_sas=\")\n",
    "\n",
    "            a += (len(\"AN_sas=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Number South Asian\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AF_sas=\")\n",
    "\n",
    "            a += (len(\"AF_sas=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Frequency South Asian\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "            a = last_field.find(\"AC_remaining=\")\n",
    "\n",
    "            a += (len(\"AC_remaining=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Count Remaining\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AN_remaining=\")\n",
    "\n",
    "            a += (len(\"AN_remaining=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Number Remaining\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AF_remaining=\")\n",
    "\n",
    "            a += (len(\"AF_remaining=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Frequency Remaining\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "            a = last_field.find(\"AC_afr=\")\n",
    "\n",
    "            a += (len(\"AC_afr=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Count African/African American\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AN_afr=\")\n",
    "\n",
    "            a += (len(\"AN_afr=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Number African/African American\": last_field[a:a+b]})\n",
    "\n",
    "            a = last_field.find(\"AF_afr=\")\n",
    "\n",
    "            a += (len(\"AF_afr=\"))\n",
    "\n",
    "            b = last_field[a:].find(\";\")\n",
    "\n",
    "            combined_row.update({\"Allele Frequency African/African American\": last_field[a:a+b]})\n",
    "\n",
    "            merged_rows.append(combined_row)\n",
    "\n",
    "            #break\n",
    "\n",
    "    #break\n",
    "    \n",
    "\n",
    "# Convert merged rows to DataFrame\n",
    "merged_df = pd.DataFrame(merged_rows)\n",
    "\n",
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d6530cb2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "46538602"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_df.iloc[len(gtex_df)-1]['end']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "cbb560b1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chromo                                           chr21\n",
       "strand                                               -\n",
       "start                                          5124755\n",
       "end                                            5124875\n",
       "genename                       CH507-9B2.3,CH507-9B2.4\n",
       "Major_Exon_Size                                    120\n",
       "count                                              202\n",
       "alt5                                                 7\n",
       "constitLevel                                     0.967\n",
       "alt5usageLevel                                   0.033\n",
       "alt5singleCount                                      7\n",
       "alt5singleLoc                                  5124674\n",
       "Alt_Exon_Size                                      201\n",
       "5.ss.Con                         chr21:5124749-5124758\n",
       "5.ss.Seq                                     CCTGTGCGT\n",
       "MaxENT                                            4.27\n",
       "5.ss.Alt                         chr21:5124668-5124677\n",
       "5.ss.Alt.Seq                                 GTGGTCAGT\n",
       "ALTMaxENT                                         4.85\n",
       "?MaxEnt                                          -0.58\n",
       "Upstream_intronSize                                149\n",
       "Downstream_intronSize                             2952\n",
       "Upperstream_intron_gcpct                      0.591241\n",
       "Exon_gcpct                                     0.53913\n",
       "Downstream_intron_gcpct                           0.54\n",
       "Alt5Exon_gcpct                                0.545918\n",
       "Alt5Downstream_intron_gcpct                       0.58\n",
       "Name: 2063, dtype: object"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_df.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "421175ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromo</th>\n",
       "      <th>strand</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>genename</th>\n",
       "      <th>Major_Exon_Size</th>\n",
       "      <th>count</th>\n",
       "      <th>alt5</th>\n",
       "      <th>constitLevel</th>\n",
       "      <th>alt5usageLevel</th>\n",
       "      <th>...</th>\n",
       "      <th>?MaxEnt</th>\n",
       "      <th>Upstream_intronSize</th>\n",
       "      <th>Downstream_intronSize</th>\n",
       "      <th>Upperstream_intron_gcpct</th>\n",
       "      <th>Exon_gcpct</th>\n",
       "      <th>Downstream_intron_gcpct</th>\n",
       "      <th>Alt5Exon_gcpct</th>\n",
       "      <th>Alt5Downstream_intron_gcpct</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr21</td>\n",
       "      <td>-</td>\n",
       "      <td>5124755</td>\n",
       "      <td>5124875</td>\n",
       "      <td>CH507-9B2.3,CH507-9B2.4</td>\n",
       "      <td>120</td>\n",
       "      <td>202</td>\n",
       "      <td>7</td>\n",
       "      <td>0.967</td>\n",
       "      <td>0.033</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.58</td>\n",
       "      <td>149</td>\n",
       "      <td>2952</td>\n",
       "      <td>0.591241</td>\n",
       "      <td>0.53913</td>\n",
       "      <td>0.54</td>\n",
       "      <td>0.545918</td>\n",
       "      <td>0.58</td>\n",
       "      <td>chr21</td>\n",
       "      <td>44943686</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromo strand    start      end                 genename  Major_Exon_Size  \\\n",
       "0  chr21      -  5124755  5124875  CH507-9B2.3,CH507-9B2.4              120   \n",
       "\n",
       "   count  alt5  constitLevel  alt5usageLevel  ...  ?MaxEnt  \\\n",
       "0    202     7         0.967           0.033  ...    -0.58   \n",
       "\n",
       "   Upstream_intronSize  Downstream_intronSize Upperstream_intron_gcpct  \\\n",
       "0                  149                   2952                 0.591241   \n",
       "\n",
       "  Exon_gcpct  Downstream_intron_gcpct Alt5Exon_gcpct  \\\n",
       "0    0.53913                     0.54       0.545918   \n",
       "\n",
       "  Alt5Downstream_intron_gcpct  Chromosome  Position  \n",
       "0                        0.58       chr21  44943686  \n",
       "\n",
       "[1 rows x 29 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "68da4666",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'chr21\\t44943686\\trs535219131\\tC\\tA\\t.\\tPASS\\tAC=4;AN=152232;AF=2.62757e-05;grpmax=amr;fafmax_faf95_max=8.87700e-05;fafmax_faf95_max_gen_anc=amr;AC_XX=0;AF_XX=0.00000;AN_XX=77854;nhomalt_XX=0;AC_XY=4;AF_XY=5.37793e-05;AN_XY=74378;nhomalt_XY=0;nhomalt=0;AC_afr_XX=0;AF_afr_XX=0.00000;AN_afr_XX=22160;nhomalt_afr_XX=0;AC_afr_XY=0;AF_afr_XY=0.00000;AN_afr_XY=19310;nhomalt_afr_XY=0;AC_afr=0;AF_afr=0.00000;AN_afr=41470;nhomalt_afr=0;AC_ami_XX=0;AF_ami_XX=0.00000;AN_ami_XX=470;nhomalt_ami_XX=0;AC_ami_XY=0;AF_ami_XY=0.00000;AN_ami_XY=442;nhomalt_ami_XY=0;AC_ami=0;AF_ami=0.00000;AN_ami=912;nhomalt_ami=0;AC_amr_XX=0;AF_amr_XX=0.00000;AN_amr_XX=6790;nhomalt_amr_XX=0;AC_amr_XY=4;AF_amr_XY=0.000471032;AN_amr_XY=8492;nhomalt_amr_XY=0;AC_amr=4;AF_amr=0.000261746;AN_amr=15282;nhomalt_amr=0;AC_asj_XX=0;AF_asj_XX=0.00000;AN_asj_XX=1864;nhomalt_asj_XX=0;AC_asj_XY=0;AF_asj_XY=0.00000;AN_asj_XY=1604;nhomalt_asj_XY=0;AC_asj=0;AF_asj=0.00000;AN_asj=3468;nhomalt_asj=0;AC_eas_XX=0;AF_eas_XX=0.00000;AN_eas_XX=2278;nhomalt_eas_XX=0;AC_eas_XY=0;AF_eas_XY=0.00000;AN_eas_XY=2922;nhomalt_eas_XY=0;AC_eas=0;AF_eas=0.00000;AN_eas=5200;nhomalt_eas=0;AC_fin_XX=0;AF_fin_XX=0.00000;AN_fin_XX=2574;nhomalt_fin_XX=0;AC_fin_XY=0;AF_fin_XY=0.00000;AN_fin_XY=8050;nhomalt_fin_XY=0;AC_fin=0;AF_fin=0.00000;AN_fin=10624;nhomalt_fin=0;AC_mid_XX=0;AF_mid_XX=0.00000;AN_mid_XX=164;nhomalt_mid_XX=0;AC_mid_XY=0;AF_mid_XY=0.00000;AN_mid_XY=152;nhomalt_mid_XY=0;AC_mid=0;AF_mid=0.00000;AN_mid=316;nhomalt_mid=0;AC_nfe_XX=0;AF_nfe_XX=0.00000;AN_nfe_XX=39354;nhomalt_nfe_XX=0;AC_nfe_XY=0;AF_nfe_XY=0.00000;AN_nfe_XY=28684;nhomalt_nfe_XY=0;AC_nfe=0;AF_nfe=0.00000;AN_nfe=68038;nhomalt_nfe=0;AC_raw=4;AF_raw=2.62650e-05;AN_raw=152294;nhomalt_raw=0;AC_remaining_XX=0;AF_remaining_XX=0.00000;AN_remaining_XX=1034;nhomalt_remaining_XX=0;AC_remaining_XY=0;AF_remaining_XY=0.00000;AN_remaining_XY=1060;nhomalt_remaining_XY=0;AC_remaining=0;AF_remaining=0.00000;AN_remaining=2094;nhomalt_remaining=0;AC_sas_XX=0;AF_sas_XX=0.00000;AN_sas_XX=1166;nhomalt_sas_XX=0;AC_sas_XY=0;AF_sas_XY=0.00000;AN_sas_XY=3662;nhomalt_sas_XY=0;AC_sas=0;AF_sas=0.00000;AN_sas=4828;nhomalt_sas=0;AC_joint_XX=0;AF_joint_XX=0.00000;AN_joint_XX=363998;nhomalt_joint_XX=0;AC_joint_XY=5;AF_joint_XY=1.19956e-05;AN_joint_XY=416820;nhomalt_joint_XY=0;AC_joint=5;AF_joint=6.40354e-06;AN_joint=780818;nhomalt_joint=0;AC_joint_afr_XX=0;AF_joint_afr_XX=0.00000;AN_joint_afr_XX=32442;nhomalt_joint_afr_XX=0;AC_joint_afr_XY=1;AF_joint_afr_XY=3.74336e-05;AN_joint_afr_XY=26714;nhomalt_joint_afr_XY=0;AC_joint_afr=1;AF_joint_afr=1.69045e-05;AN_joint_afr=59156;nhomalt_joint_afr=0;AC_joint_ami_XX=0;AF_joint_ami_XX=0.00000;AN_joint_ami_XX=470;nhomalt_joint_ami_XX=0;AC_joint_ami_XY=0;AF_joint_ami_XY=0.00000;AN_joint_ami_XY=442;nhomalt_joint_ami_XY=0;AC_joint_ami=0;AF_joint_ami=0.00000;AN_joint_ami=912;nhomalt_joint_ami=0;AC_joint_amr_XX=0;AF_joint_amr_XX=0.00000;AN_joint_amr_XX=31826;nhomalt_joint_amr_XX=0;AC_joint_amr_XY=4;AF_joint_amr_XY=0.000147124;AN_joint_amr_XY=27188;nhomalt_joint_amr_XY=0;AC_joint_amr=4;AF_joint_amr=6.77805e-05;AN_joint_amr=59014;nhomalt_joint_amr=0;AC_joint_asj_XX=0;AF_joint_asj_XX=0.00000;AN_joint_asj_XX=11700;nhomalt_joint_asj_XX=0;AC_joint_asj_XY=0;AF_joint_asj_XY=0.00000;AN_joint_asj_XY=12746;nhomalt_joint_asj_XY=0;AC_joint_asj=0;AF_joint_asj=0.00000;AN_joint_asj=24446;nhomalt_joint_asj=0;AC_joint_eas_XX=0;AF_joint_eas_XX=0.00000;AN_joint_eas_XX=20576;nhomalt_joint_eas_XX=0;AC_joint_eas_XY=0;AF_joint_eas_XY=0.00000;AN_joint_eas_XY=20694;nhomalt_joint_eas_XY=0;AC_joint_eas=0;AF_joint_eas=0.00000;AN_joint_eas=41270;nhomalt_joint_eas=0;AC_joint_fin_XX=0;AF_joint_fin_XX=0.00000;AN_joint_fin_XX=29970;nhomalt_joint_fin_XX=0;AC_joint_fin_XY=0;AF_joint_fin_XY=0.00000;AN_joint_fin_XY=33794;nhomalt_joint_fin_XY=0;AC_joint_fin=0;AF_joint_fin=0.00000;AN_joint_fin=63764;nhomalt_joint_fin=0;AC_joint_mid_XX=0;AF_joint_mid_XX=0.00000;AN_joint_mid_XX=2068;nhomalt_joint_mid_XX=0;AC_joint_mid_XY=0;AF_joint_mid_XY=0.00000;AN_joint_mid_XY=2378;nhomalt_joint_mid_XY=0;AC_joint_mid=0;AF_joint_mid=0.00000;AN_joint_mid=4446;nhomalt_joint_mid=0;AC_joint_nfe_XX=0;AF_joint_nfe_XX=0.00000;AN_joint_nfe_XX=201496;nhomalt_joint_nfe_XX=0;AC_joint_nfe_XY=0;AF_joint_nfe_XY=0.00000;AN_joint_nfe_XY=216514;nhomalt_joint_nfe_XY=0;AC_joint_nfe=0;AF_joint_nfe=0.00000;AN_joint_nfe=418010;nhomalt_joint_nfe=0;AC_joint_raw=5;AF_joint_raw=6.40141e-06;AN_joint_raw=781078;nhomalt_joint_raw=0;AC_joint_remaining_XX=0;AF_joint_remaining_XX=0.00000;AN_joint_remaining_XX=17782;nhomalt_joint_remaining_XX=0;AC_joint_remaining_XY=0;AF_joint_remaining_XY=0.00000;AN_joint_remaining_XY=17394;nhomalt_joint_remaining_XY=0;AC_joint_remaining=0;AF_joint_remaining=0.00000;AN_joint_remaining=35176;nhomalt_joint_remaining=0;AC_joint_sas_XX=0;AF_joint_sas_XX=0.00000;AN_joint_sas_XX=15668;nhomalt_joint_sas_XX=0;AC_joint_sas_XY=0;AF_joint_sas_XY=0.00000;AN_joint_sas_XY=58956;nhomalt_joint_sas_XY=0;AC_joint_sas=0;AF_joint_sas=0.00000;AN_joint_sas=74624;nhomalt_joint_sas=0;AC_grpmax=4;AF_grpmax=0.000261746;AN_grpmax=15282;nhomalt_grpmax=0;grpmax_joint=amr;AC_grpmax_joint=4;AF_grpmax_joint=6.77805e-05;AN_grpmax_joint=59014;nhomalt_grpmax_joint=0;faf95=8.14000e-06;faf95_afr=0.00000;faf95_amr=8.87700e-05;faf95_eas=0.00000;faf95_nfe=0.00000;faf95_sas=0.00000;faf99=5.14000e-06;faf99_afr=0.00000;faf99_amr=5.38600e-05;faf99_eas=0.00000;faf99_nfe=0.00000;faf99_sas=0.00000;fafmax_faf99_max=5.38600e-05;fafmax_faf99_max_gen_anc=amr;faf95_joint=1.88000e-06;faf95_joint_afr=0.00000;faf95_joint_amr=2.28100e-05;faf95_joint_eas=0.00000;faf95_joint_nfe=0.00000;faf95_joint_sas=0.00000;faf99_joint=1.36000e-06;faf99_joint_afr=0.00000;faf99_joint_amr=1.36500e-05;faf99_joint_eas=0.00000;faf99_joint_nfe=0.00000;faf99_joint_sas=0.00000;fafmax_faf95_max_joint=2.28100e-05;fafmax_faf95_max_gen_anc_joint=amr;fafmax_faf99_max_joint=1.36500e-05;fafmax_faf99_max_gen_anc_joint=amr;fafmax_data_type_joint=genomes;age_hist_het_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_het_n_smaller=0;age_hist_het_n_larger=0;age_hist_hom_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_hom_n_smaller=0;age_hist_hom_n_larger=0;FS=0.422541;MQ=60.0000;MQRankSum=0.322000;QUALapprox=4689;QD=15.3738;ReadPosRankSum=0.332000;SOR=0.645000;VarDP=305;AS_FS=1.46649;AS_MQ=60.0000;AS_MQRankSum=0.322000;AS_pab_max=1.00000;AS_QUALapprox=3819;AS_QD=16.0462;AS_ReadPosRankSum=0.267000;AS_SB_TABLE=80,57|75,60;AS_SOR=0.585501;AS_VarDP=238;inbreeding_coeff=-2.62657e-05;AS_culprit=AS_MQ;AS_VQSLOD=5.07460;allele_type=snv;n_alt_alleles=3;variant_type=multi-snv;gq_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|1|3;gq_hist_all_bin_freq=0|0|0|0|13564|11862|8608|6458|6719|5569|4372|4648|3611|2755|2355|1641|1207|906|554|1287;dp_hist_alt_bin_freq=0|0|1|0|1|1|1|0|0|0|0|0|0|0|0|0|0|0|0|0;dp_hist_alt_n_larger=0;dp_hist_all_bin_freq=0|0|206|2377|7134|13026|27782|19470|3921|898|598|336|175|77|47|24|15|11|3|5;dp_hist_all_n_larger=11;ab_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|2|1|1|0|0|0|0|0|0;cadd_raw_score=0.939960;cadd_phred=10.8800;revel_max=0.0720000;spliceai_ds_max=0.0400000;pangolin_largest_ds=0.0200000;phylop=-0.852000;sift_max=0.280000;polyphen_max=0.0380000;VRS_Allele_IDs=ga4gh:VA.lxWzltu3SB36YYhOEMbjdZIfVJywMHSx,ga4gh:VA.fzUoay62RdxOR0ZfHozcpuKzhaX5Dp9O;VRS_Starts=44943685,44943685;VRS_Ends=44943686,44943686;VRS_States=C,A;vep=A|missense_variant&splice_region_variant|MODERATE|SLX9|ENSG00000160256|Transcript|ENST00000291634|protein_coding|2/6||ENST00000291634.11:c.132C>A|ENSP00000291634.6:p.Asp44Glu|161|132|44|D/E|gaC/gaA|1||1||SNV|HGNC|HGNC:15811|YES|NM_058190.4||1||CCDS13718.1|ENSP00000291634|Q9NSI2-1|Ensembl|||PANTHER:PTHR31109&PANTHER:PTHR31109||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|ENSG00000183250|Transcript|ENST00000330551|lncRNA||||||||||1|3841|-1||SNV|HGNC|HGNC:15707||||1|||||Ensembl|||||||||||||||,A|missense_variant&splice_region_variant|MODERATE|SLX9|ENSG00000160256|Transcript|ENST00000397826|protein_coding|2/6||ENST00000397826.7:c.132C>A|ENSP00000380926.3:p.Asp44Glu|178|132|44|D/E|gaC/gaA|1||1||SNV|HGNC|HGNC:15811||||1|P1|CCDS82682.1|ENSP00000380926|Q9NSI2-2|Ensembl|||PANTHER:PTHR31109&PANTHER:PTHR31109||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|ENSG00000183250|Transcript|ENST00000397841|lncRNA||||||||||1|3773|-1||SNV|HGNC|HGNC:15707||||2|||||Ensembl|||||||||||||||,A|missense_variant&splice_region_variant|MODERATE|SLX9|ENSG00000160256|Transcript|ENST00000458015|protein_coding|2/5||ENST00000458015.1:c.132C>A|ENSP00000404964.1:p.Asp44Glu|165|132|44|D/E|gaC/gaA|1||1||SNV|HGNC|HGNC:15811||||5|||ENSP00000404964||Ensembl|||PANTHER:PTHR31109&PANTHER:PTHR31109||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|ENSG00000183250|Transcript|ENST00000615847|lncRNA||||||||||1|3749|-1||SNV|HGNC|HGNC:15707|YES|||1|||||Ensembl|||||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|ENSG00000183250|Transcript|ENST00000654166|lncRNA||||||||||1|3745|-1||SNV|HGNC|HGNC:15707|||||||||Ensembl|||||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|ENSG00000183250|Transcript|ENST00000657006|lncRNA||||||||||1|4113|-1||SNV|HGNC|HGNC:15707|||||||||Ensembl|||||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|ENSG00000183250|Transcript|ENST00000658179|lncRNA||||||||||1|3770|-1||SNV|HGNC|HGNC:15707|||||||||Ensembl|||||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|ENSG00000183250|Transcript|ENST00000660377|lncRNA||||||||||1|3745|-1||SNV|HGNC|HGNC:15707|||||||||Ensembl|||||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|ENSG00000183250|Transcript|ENST00000663249|lncRNA||||||||||1|3755|-1||SNV|HGNC|HGNC:15707|||||||||Ensembl|||||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|ENSG00000183250|Transcript|ENST00000664358|lncRNA||||||||||1|4095|-1||SNV|HGNC|HGNC:15707|||||||||Ensembl|||||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|ENSG00000183250|Transcript|ENST00000667826|lncRNA||||||||||1|3760|-1||SNV|HGNC|HGNC:15707|||||||||Ensembl|||||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|ENSG00000183250|Transcript|ENST00000690957|lncRNA||||||||||1|3768|-1||SNV|HGNC|HGNC:15707|||||||||Ensembl|||||||||||||||,A|missense_variant&splice_region_variant|MODERATE|FAM207A|85395|Transcript|NM_001316983.2|protein_coding|2/6||NM_001316983.2:c.132C>A|NP_001303912.1:p.Asp44Glu|161|132|44|D/E|gaC/gaA|1||1||SNV|EntrezGene||||||||NP_001303912.1||RefSeq|||||||||||||||,A|missense_variant&splice_region_variant|MODERATE|FAM207A|85395|Transcript|NM_001316984.2|protein_coding|2/6||NM_001316984.2:c.132C>A|NP_001303913.1:p.Asp44Glu|161|132|44|D/E|gaC/gaA|1||1||SNV|EntrezGene||||||||NP_001303913.1||RefSeq|||||||||||||||,A|missense_variant&splice_region_variant|MODERATE|FAM207A|85395|Transcript|NM_001316985.2|protein_coding|2/6||NM_001316985.2:c.132C>A|NP_001303914.1:p.Asp44Glu|161|132|44|D/E|gaC/gaA|1||1||SNV|EntrezGene||||||||NP_001303914.1||RefSeq|||||||||||||||,A|missense_variant&splice_region_variant|MODERATE|FAM207A|85395|Transcript|NM_001316986.2|protein_coding|2/5||NM_001316986.2:c.132C>A|NP_001303915.1:p.Asp44Glu|161|132|44|D/E|gaC/gaA|1||1||SNV|EntrezGene||||||||NP_001303915.1||RefSeq|||||||||||||||,A|missense_variant&splice_region_variant|MODERATE|FAM207A|85395|Transcript|NM_001316987.2|protein_coding|2/5||NM_001316987.2:c.132C>A|NP_001303916.1:p.Asp44Glu|161|132|44|D/E|gaC/gaA|1||1||SNV|EntrezGene||||||||NP_001303916.1||RefSeq|||||||||||||||,A|splice_region_variant&5_prime_UTR_variant|LOW|FAM207A|85395|Transcript|NM_001316988.2|protein_coding|2/6||NM_001316988.2:c.-145C>A||176|||||1||1||SNV|EntrezGene||||||||NP_001303917.1||RefSeq|||||||||||||||,A|missense_variant&splice_region_variant|MODERATE|FAM207A|85395|Transcript|NM_058190.4|protein_coding|2/6||NM_058190.4:c.132C>A|NP_478070.1:p.Asp44Glu|161|132|44|D/E|gaC/gaA|1||1||SNV|EntrezGene||YES|ENST00000291634.11|||||NP_478070.1||RefSeq|||||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|84536|Transcript|NR_027128.1|lncRNA||||||||||1|3773|-1||SNV|EntrezGene|HGNC:15707|YES||||||||RefSeq|||||||||||||||,A|upstream_gene_variant|MODIFIER|LINC01547|84536|Transcript|NR_027129.1|lncRNA||||||||||1|3773|-1||SNV|EntrezGene|HGNC:15707|||||||||RefSeq|||||||||||||||,A|splice_region_variant&5_prime_UTR_variant|LOW|FAM207A|85395|Transcript|XM_017028481.2|protein_coding|2/6||XM_017028481.2:c.-190C>A||342|||||1||1||SNV|EntrezGene||||||||XP_016883970.1||RefSeq|||||||||||||||,A|upstream_gene_variant|MODIFIER|LOC107985497|107985497|Transcript|XR_001755084.1|lncRNA||||||||||1|4942|1||SNV|EntrezGene||YES||||||||RefSeq|||||||||||||||\\n'"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = row.find(\"\\t\", 0 + 1)\n",
    "\n",
    "b = row.find(\"\\t\", a + 1)\n",
    "\n",
    "row[:a], row[a+1: b]\n",
    "\n",
    "row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "20ac3a95",
   "metadata": {},
   "outputs": [],
   "source": [
    "fields = line.strip().split('\\t')\n",
    "\n",
    "\n",
    "combined_row.update({\"Chromosome\": fields[0]})\n",
    "\n",
    "combined_row.update({\"Position\": fields[1]})\n",
    "\n",
    "combined_row.update({\"rsIDs\": fields[2]})\n",
    "\n",
    "combined_row.update({\"Reference\": fields[3]})\n",
    "\n",
    "combined_row.update({\"Alternate\": fields[4]})\n",
    "\n",
    "last_field = fields[-1]\n",
    "\n",
    "a = last_field.find(\"AC=\")\n",
    "\n",
    "a += len(\"AC=\")\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Count\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AN=\")\n",
    "\n",
    "a += len(\"AN=\")\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Number\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AF=\")\n",
    "\n",
    "a += len(\"AF=\")\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Frequency\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "\n",
    "a = last_field.find(\"AC_ami=\")\n",
    "\n",
    "a += (len(\"AC_ami=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Count Amish\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AN_ami=\")\n",
    "\n",
    "a += (len(\"AN_ami=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Number Amish\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AF_ami=\")\n",
    "\n",
    "a += (len(\"AF_ami=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Frequency Amish\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "a = last_field.find(\"AC_amr=\")\n",
    "\n",
    "a += (len(\"AC_amr=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Count Admixed American\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AN_amr=\")\n",
    "\n",
    "a += (len(\"AN_amr=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Number Admixed American\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AF_amr=\")\n",
    "\n",
    "a += (len(\"AF_amr=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Frequency Admixed American\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "a = last_field.find(\"AC_asj=\")\n",
    "\n",
    "a += (len(\"AC_asj=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Count Ashkenazi Jewish\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AN_asj=\")\n",
    "\n",
    "a += (len(\"AN_asj=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Number Ashkenazi Jewish\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AF_asj=\")\n",
    "\n",
    "a += (len(\"AF_asj=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Frequency Ashkenazi Jewish\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "\n",
    "a = last_field.find(\"AC_eas=\")\n",
    "\n",
    "a += (len(\"AC_eas=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Count East Asian\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AN_eas=\")\n",
    "\n",
    "a += (len(\"AN_eas=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Number East Asian\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AF_eas=\")\n",
    "\n",
    "a += (len(\"AF_eas=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Frequency East Asian\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "a = last_field.find(\"AC_fin=\")\n",
    "\n",
    "a += (len(\"AC_fin=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Count European (Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AN_fin=\")\n",
    "\n",
    "a += (len(\"AN_fin=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Number European (Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AF_fin=\")\n",
    "\n",
    "a += (len(\"AF_fin=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Frequency European (Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "a = last_field.find(\"AC_nfe=\")\n",
    "\n",
    "a += (len(\"AC_nfe=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Count European (non-Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AN_nfe=\")\n",
    "\n",
    "a += (len(\"AN_nfe=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Number European (non-Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AF_nfe=\")\n",
    "\n",
    "a += (len(\"AF_nfe=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Frequency European (non-Finnish)\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "a = last_field.find(\"AC_sas=\")\n",
    "\n",
    "a += (len(\"AC_sas=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Count South Asian\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AN_sas=\")\n",
    "\n",
    "a += (len(\"AN_sas=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Number South Asian\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AF_sas=\")\n",
    "\n",
    "a += (len(\"AF_sas=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Frequency South Asian\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "a = last_field.find(\"AC_remaining=\")\n",
    "\n",
    "a += (len(\"AC_remaining=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Count Remaining\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AN_remaining=\")\n",
    "\n",
    "a += (len(\"AN_remaining=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Number Remaining\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AF_remaining=\")\n",
    "\n",
    "a += (len(\"AF_remaining=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Frequency Remaining\": last_field[a:a+b]})\n",
    "\n",
    "\n",
    "a = last_field.find(\"AC_afr=\")\n",
    "\n",
    "a += (len(\"AC_afr=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Count African/African American\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AN_afr=\")\n",
    "\n",
    "a += (len(\"AN_afr=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Number African/African American\": last_field[a:a+b]})\n",
    "\n",
    "a = last_field.find(\"AF_afr=\")\n",
    "\n",
    "a += (len(\"AF_afr=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "combined_row.update({\"Allele Frequency African/African American\": last_field[a:a+b]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "5a623732",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['AC=2',\n",
       " 'AN=152308',\n",
       " 'AF=1.31313e-05',\n",
       " 'grpmax=afr',\n",
       " 'fafmax_faf95_max=7.99000e-06',\n",
       " 'fafmax_faf95_max_gen_anc=afr',\n",
       " 'AC_XX=0',\n",
       " 'AF_XX=0.00000',\n",
       " 'AN_XX=77890',\n",
       " 'nhomalt_XX=0',\n",
       " 'AC_XY=2',\n",
       " 'AF_XY=2.68752e-05',\n",
       " 'AN_XY=74418',\n",
       " 'nhomalt_XY=0',\n",
       " 'nhomalt=0',\n",
       " 'AC_afr_XX=0',\n",
       " 'AF_afr_XX=0.00000',\n",
       " 'AN_afr_XX=22168',\n",
       " 'nhomalt_afr_XX=0',\n",
       " 'AC_afr_XY=2',\n",
       " 'AF_afr_XY=0.000103541',\n",
       " 'AN_afr_XY=19316',\n",
       " 'nhomalt_afr_XY=0',\n",
       " 'AC_afr=2',\n",
       " 'AF_afr=4.82114e-05',\n",
       " 'AN_afr=41484',\n",
       " 'nhomalt_afr=0',\n",
       " 'AC_ami_XX=0',\n",
       " 'AF_ami_XX=0.00000',\n",
       " 'AN_ami_XX=470',\n",
       " 'nhomalt_ami_XX=0',\n",
       " 'AC_ami_XY=0',\n",
       " 'AF_ami_XY=0.00000',\n",
       " 'AN_ami_XY=442',\n",
       " 'nhomalt_ami_XY=0',\n",
       " 'AC_ami=0',\n",
       " 'AF_ami=0.00000',\n",
       " 'AN_ami=912',\n",
       " 'nhomalt_ami=0',\n",
       " 'AC_amr_XX=0',\n",
       " 'AF_amr_XX=0.00000',\n",
       " 'AN_amr_XX=6792',\n",
       " 'nhomalt_amr_XX=0',\n",
       " 'AC_amr_XY=0',\n",
       " 'AF_amr_XY=0.00000',\n",
       " 'AN_amr_XY=8502',\n",
       " 'nhomalt_amr_XY=0',\n",
       " 'AC_amr=0',\n",
       " 'AF_amr=0.00000',\n",
       " 'AN_amr=15294',\n",
       " 'nhomalt_amr=0',\n",
       " 'AC_asj_XX=0',\n",
       " 'AF_asj_XX=0.00000',\n",
       " 'AN_asj_XX=1868',\n",
       " 'nhomalt_asj_XX=0',\n",
       " 'AC_asj_XY=0',\n",
       " 'AF_asj_XY=0.00000',\n",
       " 'AN_asj_XY=1604',\n",
       " 'nhomalt_asj_XY=0',\n",
       " 'AC_asj=0',\n",
       " 'AF_asj=0.00000',\n",
       " 'AN_asj=3472',\n",
       " 'nhomalt_asj=0',\n",
       " 'AC_eas_XX=0',\n",
       " 'AF_eas_XX=0.00000',\n",
       " 'AN_eas_XX=2282',\n",
       " 'nhomalt_eas_XX=0',\n",
       " 'AC_eas_XY=0',\n",
       " 'AF_eas_XY=0.00000',\n",
       " 'AN_eas_XY=2926',\n",
       " 'nhomalt_eas_XY=0',\n",
       " 'AC_eas=0',\n",
       " 'AF_eas=0.00000',\n",
       " 'AN_eas=5208',\n",
       " 'nhomalt_eas=0',\n",
       " 'AC_fin_XX=0',\n",
       " 'AF_fin_XX=0.00000',\n",
       " 'AN_fin_XX=2574',\n",
       " 'nhomalt_fin_XX=0',\n",
       " 'AC_fin_XY=0',\n",
       " 'AF_fin_XY=0.00000',\n",
       " 'AN_fin_XY=8058',\n",
       " 'nhomalt_fin_XY=0',\n",
       " 'AC_fin=0',\n",
       " 'AF_fin=0.00000',\n",
       " 'AN_fin=10632',\n",
       " 'nhomalt_fin=0',\n",
       " 'AC_mid_XX=0',\n",
       " 'AF_mid_XX=0.00000',\n",
       " 'AN_mid_XX=164',\n",
       " 'nhomalt_mid_XX=0',\n",
       " 'AC_mid_XY=0',\n",
       " 'AF_mid_XY=0.00000',\n",
       " 'AN_mid_XY=152',\n",
       " 'nhomalt_mid_XY=0',\n",
       " 'AC_mid=0',\n",
       " 'AF_mid=0.00000',\n",
       " 'AN_mid=316',\n",
       " 'nhomalt_mid=0',\n",
       " 'AC_nfe_XX=0',\n",
       " 'AF_nfe_XX=0.00000',\n",
       " 'AN_nfe_XX=39372',\n",
       " 'nhomalt_nfe_XX=0',\n",
       " 'AC_nfe_XY=0',\n",
       " 'AF_nfe_XY=0.00000',\n",
       " 'AN_nfe_XY=28686',\n",
       " 'nhomalt_nfe_XY=0',\n",
       " 'AC_nfe=0',\n",
       " 'AF_nfe=0.00000',\n",
       " 'AN_nfe=68058',\n",
       " 'nhomalt_nfe=0',\n",
       " 'AC_raw=4',\n",
       " 'AF_raw=2.62619e-05',\n",
       " 'AN_raw=152312',\n",
       " 'nhomalt_raw=0',\n",
       " 'AC_remaining_XX=0',\n",
       " 'AF_remaining_XX=0.00000',\n",
       " 'AN_remaining_XX=1034',\n",
       " 'nhomalt_remaining_XX=0',\n",
       " 'AC_remaining_XY=0',\n",
       " 'AF_remaining_XY=0.00000',\n",
       " 'AN_remaining_XY=1060',\n",
       " 'nhomalt_remaining_XY=0',\n",
       " 'AC_remaining=0',\n",
       " 'AF_remaining=0.00000',\n",
       " 'AN_remaining=2094',\n",
       " 'nhomalt_remaining=0',\n",
       " 'AC_sas_XX=0',\n",
       " 'AF_sas_XX=0.00000',\n",
       " 'AN_sas_XX=1166',\n",
       " 'nhomalt_sas_XX=0',\n",
       " 'AC_sas_XY=0',\n",
       " 'AF_sas_XY=0.00000',\n",
       " 'AN_sas_XY=3672',\n",
       " 'nhomalt_sas_XY=0',\n",
       " 'AC_sas=0',\n",
       " 'AF_sas=0.00000',\n",
       " 'AN_sas=4838',\n",
       " 'nhomalt_sas=0',\n",
       " 'AC_joint_XX=0',\n",
       " 'AF_joint_XX=0.00000',\n",
       " 'AN_joint_XX=77890',\n",
       " 'nhomalt_joint_XX=0',\n",
       " 'AC_joint_XY=2',\n",
       " 'AF_joint_XY=2.68752e-05',\n",
       " 'AN_joint_XY=74418',\n",
       " 'nhomalt_joint_XY=0',\n",
       " 'AC_joint=2',\n",
       " 'AF_joint=1.31313e-05',\n",
       " 'AN_joint=152308',\n",
       " 'nhomalt_joint=0',\n",
       " 'AC_joint_afr_XX=0',\n",
       " 'AF_joint_afr_XX=0.00000',\n",
       " 'AN_joint_afr_XX=22168',\n",
       " 'nhomalt_joint_afr_XX=0',\n",
       " 'AC_joint_afr_XY=2',\n",
       " 'AF_joint_afr_XY=0.000103541',\n",
       " 'AN_joint_afr_XY=19316',\n",
       " 'nhomalt_joint_afr_XY=0',\n",
       " 'AC_joint_afr=2',\n",
       " 'AF_joint_afr=4.82114e-05',\n",
       " 'AN_joint_afr=41484',\n",
       " 'nhomalt_joint_afr=0',\n",
       " 'AC_joint_ami_XX=0',\n",
       " 'AF_joint_ami_XX=0.00000',\n",
       " 'AN_joint_ami_XX=470',\n",
       " 'nhomalt_joint_ami_XX=0',\n",
       " 'AC_joint_ami_XY=0',\n",
       " 'AF_joint_ami_XY=0.00000',\n",
       " 'AN_joint_ami_XY=442',\n",
       " 'nhomalt_joint_ami_XY=0',\n",
       " 'AC_joint_ami=0',\n",
       " 'AF_joint_ami=0.00000',\n",
       " 'AN_joint_ami=912',\n",
       " 'nhomalt_joint_ami=0',\n",
       " 'AC_joint_amr_XX=0',\n",
       " 'AF_joint_amr_XX=0.00000',\n",
       " 'AN_joint_amr_XX=6792',\n",
       " 'nhomalt_joint_amr_XX=0',\n",
       " 'AC_joint_amr_XY=0',\n",
       " 'AF_joint_amr_XY=0.00000',\n",
       " 'AN_joint_amr_XY=8502',\n",
       " 'nhomalt_joint_amr_XY=0',\n",
       " 'AC_joint_amr=0',\n",
       " 'AF_joint_amr=0.00000',\n",
       " 'AN_joint_amr=15294',\n",
       " 'nhomalt_joint_amr=0',\n",
       " 'AC_joint_asj_XX=0',\n",
       " 'AF_joint_asj_XX=0.00000',\n",
       " 'AN_joint_asj_XX=1868',\n",
       " 'nhomalt_joint_asj_XX=0',\n",
       " 'AC_joint_asj_XY=0',\n",
       " 'AF_joint_asj_XY=0.00000',\n",
       " 'AN_joint_asj_XY=1604',\n",
       " 'nhomalt_joint_asj_XY=0',\n",
       " 'AC_joint_asj=0',\n",
       " 'AF_joint_asj=0.00000',\n",
       " 'AN_joint_asj=3472',\n",
       " 'nhomalt_joint_asj=0',\n",
       " 'AC_joint_eas_XX=0',\n",
       " 'AF_joint_eas_XX=0.00000',\n",
       " 'AN_joint_eas_XX=2282',\n",
       " 'nhomalt_joint_eas_XX=0',\n",
       " 'AC_joint_eas_XY=0',\n",
       " 'AF_joint_eas_XY=0.00000',\n",
       " 'AN_joint_eas_XY=2926',\n",
       " 'nhomalt_joint_eas_XY=0',\n",
       " 'AC_joint_eas=0',\n",
       " 'AF_joint_eas=0.00000',\n",
       " 'AN_joint_eas=5208',\n",
       " 'nhomalt_joint_eas=0',\n",
       " 'AC_joint_fin_XX=0',\n",
       " 'AF_joint_fin_XX=0.00000',\n",
       " 'AN_joint_fin_XX=2574',\n",
       " 'nhomalt_joint_fin_XX=0',\n",
       " 'AC_joint_fin_XY=0',\n",
       " 'AF_joint_fin_XY=0.00000',\n",
       " 'AN_joint_fin_XY=8058',\n",
       " 'nhomalt_joint_fin_XY=0',\n",
       " 'AC_joint_fin=0',\n",
       " 'AF_joint_fin=0.00000',\n",
       " 'AN_joint_fin=10632',\n",
       " 'nhomalt_joint_fin=0',\n",
       " 'AC_joint_mid_XX=0',\n",
       " 'AF_joint_mid_XX=0.00000',\n",
       " 'AN_joint_mid_XX=164',\n",
       " 'nhomalt_joint_mid_XX=0',\n",
       " 'AC_joint_mid_XY=0',\n",
       " 'AF_joint_mid_XY=0.00000',\n",
       " 'AN_joint_mid_XY=152',\n",
       " 'nhomalt_joint_mid_XY=0',\n",
       " 'AC_joint_mid=0',\n",
       " 'AF_joint_mid=0.00000',\n",
       " 'AN_joint_mid=316',\n",
       " 'nhomalt_joint_mid=0',\n",
       " 'AC_joint_nfe_XX=0',\n",
       " 'AF_joint_nfe_XX=0.00000',\n",
       " 'AN_joint_nfe_XX=39372',\n",
       " 'nhomalt_joint_nfe_XX=0',\n",
       " 'AC_joint_nfe_XY=0',\n",
       " 'AF_joint_nfe_XY=0.00000',\n",
       " 'AN_joint_nfe_XY=28686',\n",
       " 'nhomalt_joint_nfe_XY=0',\n",
       " 'AC_joint_nfe=0',\n",
       " 'AF_joint_nfe=0.00000',\n",
       " 'AN_joint_nfe=68058',\n",
       " 'nhomalt_joint_nfe=0',\n",
       " 'AC_joint_raw=4',\n",
       " 'AF_joint_raw=2.62619e-05',\n",
       " 'AN_joint_raw=152312',\n",
       " 'nhomalt_joint_raw=0',\n",
       " 'AC_joint_remaining_XX=0',\n",
       " 'AF_joint_remaining_XX=0.00000',\n",
       " 'AN_joint_remaining_XX=1034',\n",
       " 'nhomalt_joint_remaining_XX=0',\n",
       " 'AC_joint_remaining_XY=0',\n",
       " 'AF_joint_remaining_XY=0.00000',\n",
       " 'AN_joint_remaining_XY=1060',\n",
       " 'nhomalt_joint_remaining_XY=0',\n",
       " 'AC_joint_remaining=0',\n",
       " 'AF_joint_remaining=0.00000',\n",
       " 'AN_joint_remaining=2094',\n",
       " 'nhomalt_joint_remaining=0',\n",
       " 'AC_joint_sas_XX=0',\n",
       " 'AF_joint_sas_XX=0.00000',\n",
       " 'AN_joint_sas_XX=1166',\n",
       " 'nhomalt_joint_sas_XX=0',\n",
       " 'AC_joint_sas_XY=0',\n",
       " 'AF_joint_sas_XY=0.00000',\n",
       " 'AN_joint_sas_XY=3672',\n",
       " 'nhomalt_joint_sas_XY=0',\n",
       " 'AC_joint_sas=0',\n",
       " 'AF_joint_sas=0.00000',\n",
       " 'AN_joint_sas=4838',\n",
       " 'nhomalt_joint_sas=0',\n",
       " 'AC_grpmax=2',\n",
       " 'AF_grpmax=4.82114e-05',\n",
       " 'AN_grpmax=41484',\n",
       " 'nhomalt_grpmax=0',\n",
       " 'grpmax_joint=afr',\n",
       " 'AC_grpmax_joint=2',\n",
       " 'AF_grpmax_joint=4.82114e-05',\n",
       " 'AN_grpmax_joint=41484',\n",
       " 'nhomalt_grpmax_joint=0',\n",
       " 'faf95=2.18000e-06',\n",
       " 'faf95_afr=7.99000e-06',\n",
       " 'faf95_amr=0.00000',\n",
       " 'faf95_eas=0.00000',\n",
       " 'faf95_nfe=0.00000',\n",
       " 'faf95_sas=0.00000',\n",
       " 'faf99=8.20000e-07',\n",
       " 'faf99_afr=2.99000e-06',\n",
       " 'faf99_amr=0.00000',\n",
       " 'faf99_eas=0.00000',\n",
       " 'faf99_nfe=0.00000',\n",
       " 'faf99_sas=0.00000',\n",
       " 'fafmax_faf99_max=2.99000e-06',\n",
       " 'fafmax_faf99_max_gen_anc=afr',\n",
       " 'faf95_joint=2.18000e-06',\n",
       " 'faf95_joint_afr=7.99000e-06',\n",
       " 'faf95_joint_amr=0.00000',\n",
       " 'faf95_joint_eas=0.00000',\n",
       " 'faf95_joint_nfe=0.00000',\n",
       " 'faf95_joint_sas=0.00000',\n",
       " 'faf99_joint=8.20000e-07',\n",
       " 'faf99_joint_afr=2.99000e-06',\n",
       " 'faf99_joint_amr=0.00000',\n",
       " 'faf99_joint_eas=0.00000',\n",
       " 'faf99_joint_nfe=0.00000',\n",
       " 'faf99_joint_sas=0.00000',\n",
       " 'fafmax_faf95_max_joint=7.99000e-06',\n",
       " 'fafmax_faf95_max_gen_anc_joint=afr',\n",
       " 'fafmax_faf99_max_joint=2.99000e-06',\n",
       " 'fafmax_faf99_max_gen_anc_joint=afr',\n",
       " 'fafmax_data_type_joint=genomes',\n",
       " 'age_hist_het_bin_freq=0|0|1|1|0|0|0|0|0|0',\n",
       " 'age_hist_het_n_smaller=0',\n",
       " 'age_hist_het_n_larger=0',\n",
       " 'age_hist_hom_bin_freq=0|0|0|0|0|0|0|0|0|0',\n",
       " 'age_hist_hom_n_smaller=0',\n",
       " 'age_hist_hom_n_larger=0',\n",
       " 'FS=2.37367',\n",
       " 'MQ=49.0312',\n",
       " 'MQRankSum=-1.25400',\n",
       " 'QUALapprox=1760',\n",
       " 'QD=4.32432',\n",
       " 'ReadPosRankSum=-0.509000',\n",
       " 'SOR=0.950000',\n",
       " 'VarDP=407',\n",
       " 'AS_FS=2.37367',\n",
       " 'AS_MQ=49.0312',\n",
       " 'AS_MQRankSum=-1.25400',\n",
       " 'AS_pab_max=1.11230e-05',\n",
       " 'AS_QUALapprox=1760',\n",
       " 'AS_QD=4.32432',\n",
       " 'AS_ReadPosRankSum=-0.509000',\n",
       " 'AS_SB_TABLE=171,164|41,31',\n",
       " 'AS_SOR=0.949839',\n",
       " 'AS_VarDP=407',\n",
       " 'inbreeding_coeff=-2.62626e-05',\n",
       " 'AS_culprit=AS_QD',\n",
       " 'AS_VQSLOD=-2.15890',\n",
       " 'negative_train_site',\n",
       " 'allele_type=snv',\n",
       " 'n_alt_alleles=1',\n",
       " 'variant_type=snv',\n",
       " 'segdup',\n",
       " 'gq_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|2',\n",
       " 'gq_hist_all_bin_freq=0|0|0|0|475|319|375|355|461|530|583|689|1423|814|992|1015|1192|1340|1449|64142',\n",
       " 'dp_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|1|0|0|0|0|0|1|0',\n",
       " 'dp_hist_alt_n_larger=0',\n",
       " 'dp_hist_all_bin_freq=0|0|4|16|59|176|732|1896|2019|1864|2486|3271|4109|4922|5341|5731|5564|5478|5394|5925',\n",
       " 'dp_hist_all_n_larger=21167',\n",
       " 'ab_hist_alt_bin_freq=0|0|0|0|2|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0',\n",
       " 'cadd_raw_score=0.289799',\n",
       " 'cadd_phred=4.17000',\n",
       " 'pangolin_largest_ds=0.00000',\n",
       " 'phylop=0.258000',\n",
       " 'VRS_Allele_IDs=ga4gh:VA.r54fNO45KZ5K7ugJ4orP7aaKfvCqdDCG,ga4gh:VA.9QBimEQMWMbgcJJKF2oE4derPs05JfN_',\n",
       " 'VRS_Starts=10000002,10000002',\n",
       " 'VRS_Ends=10000003,10000003',\n",
       " 'VRS_States=A,T',\n",
       " 'vep=T|upstream_gene_variant|MODIFIER||ENSG00000270533|Transcript|ENST00000604687|transcribed_processed_pseudogene||||||||||1|1970|-1||SNV|||YES||||||||Ensembl|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER||ENSG00000270533|Transcript|ENST00000625012|processed_transcript||3/10|ENST00000625012.1:n.449-1955T>A|||||||1||-1||SNV||||||1|||||Ensembl|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_001754952.1|lncRNA||5/6|XR_001754952.1:n.909-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_001754953.1|lncRNA||4/5|XR_001754953.1:n.788-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_001754954.1|lncRNA||4/5|XR_001754954.1:n.841-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_002958652.1|lncRNA||4/5|XR_002958652.1:n.797-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951164.2|lncRNA||7/8|XR_951164.2:n.1140-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951170.2|lncRNA||5/6|XR_951170.2:n.867-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951171.2|lncRNA||4/5|XR_951171.2:n.696-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951172.1|lncRNA||4/5|XR_951172.1:n.490-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951173.3|lncRNA||1/2|XR_951173.3:n.253-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951174.3|lncRNA||2/3|XR_951174.3:n.377-1955T>A|||||||1||-1||SNV|EntrezGene||YES||||||||RefSeq|||||||||||||||']"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "last_field"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "5c89a954",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'2'"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = fields[-1].find(\"AC=\")\n",
    "\n",
    "a += len(\"AC=\")\n",
    "\n",
    "b = fields[-1][a:].find(\";\")\n",
    "\n",
    "fields[-1][a:a+b]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "3459c359",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'chromo': 'chr21',\n",
       " 'strand': '-',\n",
       " 'start': 5124755,\n",
       " 'end': 5124875,\n",
       " 'genename': 'CH507-9B2.3,CH507-9B2.4',\n",
       " 'Major_Exon_Size': 120,\n",
       " 'count': 202,\n",
       " 'alt5': 7,\n",
       " 'constitLevel': 0.967,\n",
       " 'alt5usageLevel': 0.033,\n",
       " 'alt5singleCount': 7,\n",
       " 'alt5singleLoc': 5124674,\n",
       " 'Alt_Exon_Size': 201,\n",
       " '5.ss.Con': 'chr21:5124749-5124758',\n",
       " '5.ss.Seq': 'CCTGTGCGT',\n",
       " 'MaxENT': 4.27,\n",
       " '5.ss.Alt': 'chr21:5124668-5124677',\n",
       " '5.ss.Alt.Seq': 'GTGGTCAGT',\n",
       " 'ALTMaxENT': 4.85,\n",
       " '?MaxEnt': -0.58,\n",
       " 'Upstream_intronSize': 149,\n",
       " 'Downstream_intronSize': 2952,\n",
       " 'Upperstream_intron_gcpct': 0.591241,\n",
       " 'Exon_gcpct': 0.53913,\n",
       " 'Downstream_intron_gcpct': 0.54,\n",
       " 'Alt5Exon_gcpct': 0.545918,\n",
       " 'Alt5Downstream_intron_gcpct': 0.58,\n",
       " 'Chromosome': 'chr21',\n",
       " 'Position': '10000003',\n",
       " 'Allele Count': '2',\n",
       " 'Allele Number': '152308',\n",
       " 'Allele Frequency': '1.31313e-05',\n",
       " 'rsIDs': 'rs1473816357',\n",
       " 'Reference': 'A',\n",
       " 'Alternate': 'T',\n",
       " 'Allele Count Amish': '0',\n",
       " 'Allele Number Amish': '912',\n",
       " 'Allele Frequency Amish': '0.00000',\n",
       " 'Allele Count Admixed American': '0',\n",
       " 'Allele Number Admixed American': '15294',\n",
       " 'Allele Frequency Admixed American': '0.00000',\n",
       " 'Allele Count African/African American': '2',\n",
       " 'Allele Number African/African American': '41484',\n",
       " 'Allele Frequency African/African American': '4.82114e-05',\n",
       " 'Allele Count Ashkenazi Jewish': '0',\n",
       " 'Allele Number Ashkenazi Jewish': '3472',\n",
       " 'Allele Frequency Ashkenazi Jewish': '0.00000',\n",
       " 'Allele Count East Asian': '0',\n",
       " 'Allele Number East Asian': '5208',\n",
       " 'Allele Frequency East Asian': '0.00000',\n",
       " 'Allele Count European (Finnish)': '0',\n",
       " 'Allele Number European (Finnish)': '10632',\n",
       " 'Allele Frequency European (Finnish)': '0.00000',\n",
       " 'Allele Count Middle Eastern': '0',\n",
       " 'Allele Number Middle Eastern': '316',\n",
       " 'Allele Frequency Middle Eastern': '0.00000',\n",
       " 'Allele Count European (non-Finnish)': '0',\n",
       " 'Allele Number European (non-Finnish)': '68058',\n",
       " 'Allele Frequency European (non-Finnish)': '0.00000',\n",
       " 'Allele Frequency South Asian': '0.00000',\n",
       " 'Allele Count European South Asian': '0',\n",
       " 'Allele Number European South Asian': '4838',\n",
       " 'Allele Frequency European South Asian': '0.00000',\n",
       " 'Allele Count South Asian': '0',\n",
       " 'Allele Number South Asian': '4838',\n",
       " 'Allele Count Remaining': '0',\n",
       " 'Allele Number Remaining': '2094',\n",
       " 'Allele Frequency Remaining': '0.00000'}"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "51aaffcf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0'"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = last_field.find(\"AC_ami=\")\n",
    "\n",
    "a += (len(\"AC_ami=\"))\n",
    "\n",
    "b = last_field[a:].find(\";\")\n",
    "\n",
    "last_field[a:a+b]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "d08acc51",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'AC=2;AN=152308;AF=1.31313e-05;grpmax=afr;fafmax_faf95_max=7.99000e-06;fafmax_faf95_max_gen_anc=afr;AC_XX=0;AF_XX=0.00000;AN_XX=77890;nhomalt_XX=0;AC_XY=2;AF_XY=2.68752e-05;AN_XY=74418;nhomalt_XY=0;nhomalt=0;AC_afr_XX=0;AF_afr_XX=0.00000;AN_afr_XX=22168;nhomalt_afr_XX=0;AC_afr_XY=2;AF_afr_XY=0.000103541;AN_afr_XY=19316;nhomalt_afr_XY=0;AC_afr=2;AF_afr=4.82114e-05;AN_afr=41484;nhomalt_afr=0;AC_ami_XX=0;AF_ami_XX=0.00000;AN_ami_XX=470;nhomalt_ami_XX=0;AC_ami_XY=0;AF_ami_XY=0.00000;AN_ami_XY=442;nhomalt_ami_XY=0;AC_ami=0;AF_ami=0.00000;AN_ami=912;nhomalt_ami=0;AC_amr_XX=0;AF_amr_XX=0.00000;AN_amr_XX=6792;nhomalt_amr_XX=0;AC_amr_XY=0;AF_amr_XY=0.00000;AN_amr_XY=8502;nhomalt_amr_XY=0;AC_amr=0;AF_amr=0.00000;AN_amr=15294;nhomalt_amr=0;AC_asj_XX=0;AF_asj_XX=0.00000;AN_asj_XX=1868;nhomalt_asj_XX=0;AC_asj_XY=0;AF_asj_XY=0.00000;AN_asj_XY=1604;nhomalt_asj_XY=0;AC_asj=0;AF_asj=0.00000;AN_asj=3472;nhomalt_asj=0;AC_eas_XX=0;AF_eas_XX=0.00000;AN_eas_XX=2282;nhomalt_eas_XX=0;AC_eas_XY=0;AF_eas_XY=0.00000;AN_eas_XY=2926;nhomalt_eas_XY=0;AC_eas=0;AF_eas=0.00000;AN_eas=5208;nhomalt_eas=0;AC_fin_XX=0;AF_fin_XX=0.00000;AN_fin_XX=2574;nhomalt_fin_XX=0;AC_fin_XY=0;AF_fin_XY=0.00000;AN_fin_XY=8058;nhomalt_fin_XY=0;AC_fin=0;AF_fin=0.00000;AN_fin=10632;nhomalt_fin=0;AC_mid_XX=0;AF_mid_XX=0.00000;AN_mid_XX=164;nhomalt_mid_XX=0;AC_mid_XY=0;AF_mid_XY=0.00000;AN_mid_XY=152;nhomalt_mid_XY=0;AC_mid=0;AF_mid=0.00000;AN_mid=316;nhomalt_mid=0;AC_nfe_XX=0;AF_nfe_XX=0.00000;AN_nfe_XX=39372;nhomalt_nfe_XX=0;AC_nfe_XY=0;AF_nfe_XY=0.00000;AN_nfe_XY=28686;nhomalt_nfe_XY=0;AC_nfe=0;AF_nfe=0.00000;AN_nfe=68058;nhomalt_nfe=0;AC_raw=4;AF_raw=2.62619e-05;AN_raw=152312;nhomalt_raw=0;AC_remaining_XX=0;AF_remaining_XX=0.00000;AN_remaining_XX=1034;nhomalt_remaining_XX=0;AC_remaining_XY=0;AF_remaining_XY=0.00000;AN_remaining_XY=1060;nhomalt_remaining_XY=0;AC_remaining=0;AF_remaining=0.00000;AN_remaining=2094;nhomalt_remaining=0;AC_sas_XX=0;AF_sas_XX=0.00000;AN_sas_XX=1166;nhomalt_sas_XX=0;AC_sas_XY=0;AF_sas_XY=0.00000;AN_sas_XY=3672;nhomalt_sas_XY=0;AC_sas=0;AF_sas=0.00000;AN_sas=4838;nhomalt_sas=0;AC_joint_XX=0;AF_joint_XX=0.00000;AN_joint_XX=77890;nhomalt_joint_XX=0;AC_joint_XY=2;AF_joint_XY=2.68752e-05;AN_joint_XY=74418;nhomalt_joint_XY=0;AC_joint=2;AF_joint=1.31313e-05;AN_joint=152308;nhomalt_joint=0;AC_joint_afr_XX=0;AF_joint_afr_XX=0.00000;AN_joint_afr_XX=22168;nhomalt_joint_afr_XX=0;AC_joint_afr_XY=2;AF_joint_afr_XY=0.000103541;AN_joint_afr_XY=19316;nhomalt_joint_afr_XY=0;AC_joint_afr=2;AF_joint_afr=4.82114e-05;AN_joint_afr=41484;nhomalt_joint_afr=0;AC_joint_ami_XX=0;AF_joint_ami_XX=0.00000;AN_joint_ami_XX=470;nhomalt_joint_ami_XX=0;AC_joint_ami_XY=0;AF_joint_ami_XY=0.00000;AN_joint_ami_XY=442;nhomalt_joint_ami_XY=0;AC_joint_ami=0;AF_joint_ami=0.00000;AN_joint_ami=912;nhomalt_joint_ami=0;AC_joint_amr_XX=0;AF_joint_amr_XX=0.00000;AN_joint_amr_XX=6792;nhomalt_joint_amr_XX=0;AC_joint_amr_XY=0;AF_joint_amr_XY=0.00000;AN_joint_amr_XY=8502;nhomalt_joint_amr_XY=0;AC_joint_amr=0;AF_joint_amr=0.00000;AN_joint_amr=15294;nhomalt_joint_amr=0;AC_joint_asj_XX=0;AF_joint_asj_XX=0.00000;AN_joint_asj_XX=1868;nhomalt_joint_asj_XX=0;AC_joint_asj_XY=0;AF_joint_asj_XY=0.00000;AN_joint_asj_XY=1604;nhomalt_joint_asj_XY=0;AC_joint_asj=0;AF_joint_asj=0.00000;AN_joint_asj=3472;nhomalt_joint_asj=0;AC_joint_eas_XX=0;AF_joint_eas_XX=0.00000;AN_joint_eas_XX=2282;nhomalt_joint_eas_XX=0;AC_joint_eas_XY=0;AF_joint_eas_XY=0.00000;AN_joint_eas_XY=2926;nhomalt_joint_eas_XY=0;AC_joint_eas=0;AF_joint_eas=0.00000;AN_joint_eas=5208;nhomalt_joint_eas=0;AC_joint_fin_XX=0;AF_joint_fin_XX=0.00000;AN_joint_fin_XX=2574;nhomalt_joint_fin_XX=0;AC_joint_fin_XY=0;AF_joint_fin_XY=0.00000;AN_joint_fin_XY=8058;nhomalt_joint_fin_XY=0;AC_joint_fin=0;AF_joint_fin=0.00000;AN_joint_fin=10632;nhomalt_joint_fin=0;AC_joint_mid_XX=0;AF_joint_mid_XX=0.00000;AN_joint_mid_XX=164;nhomalt_joint_mid_XX=0;AC_joint_mid_XY=0;AF_joint_mid_XY=0.00000;AN_joint_mid_XY=152;nhomalt_joint_mid_XY=0;AC_joint_mid=0;AF_joint_mid=0.00000;AN_joint_mid=316;nhomalt_joint_mid=0;AC_joint_nfe_XX=0;AF_joint_nfe_XX=0.00000;AN_joint_nfe_XX=39372;nhomalt_joint_nfe_XX=0;AC_joint_nfe_XY=0;AF_joint_nfe_XY=0.00000;AN_joint_nfe_XY=28686;nhomalt_joint_nfe_XY=0;AC_joint_nfe=0;AF_joint_nfe=0.00000;AN_joint_nfe=68058;nhomalt_joint_nfe=0;AC_joint_raw=4;AF_joint_raw=2.62619e-05;AN_joint_raw=152312;nhomalt_joint_raw=0;AC_joint_remaining_XX=0;AF_joint_remaining_XX=0.00000;AN_joint_remaining_XX=1034;nhomalt_joint_remaining_XX=0;AC_joint_remaining_XY=0;AF_joint_remaining_XY=0.00000;AN_joint_remaining_XY=1060;nhomalt_joint_remaining_XY=0;AC_joint_remaining=0;AF_joint_remaining=0.00000;AN_joint_remaining=2094;nhomalt_joint_remaining=0;AC_joint_sas_XX=0;AF_joint_sas_XX=0.00000;AN_joint_sas_XX=1166;nhomalt_joint_sas_XX=0;AC_joint_sas_XY=0;AF_joint_sas_XY=0.00000;AN_joint_sas_XY=3672;nhomalt_joint_sas_XY=0;AC_joint_sas=0;AF_joint_sas=0.00000;AN_joint_sas=4838;nhomalt_joint_sas=0;AC_grpmax=2;AF_grpmax=4.82114e-05;AN_grpmax=41484;nhomalt_grpmax=0;grpmax_joint=afr;AC_grpmax_joint=2;AF_grpmax_joint=4.82114e-05;AN_grpmax_joint=41484;nhomalt_grpmax_joint=0;faf95=2.18000e-06;faf95_afr=7.99000e-06;faf95_amr=0.00000;faf95_eas=0.00000;faf95_nfe=0.00000;faf95_sas=0.00000;faf99=8.20000e-07;faf99_afr=2.99000e-06;faf99_amr=0.00000;faf99_eas=0.00000;faf99_nfe=0.00000;faf99_sas=0.00000;fafmax_faf99_max=2.99000e-06;fafmax_faf99_max_gen_anc=afr;faf95_joint=2.18000e-06;faf95_joint_afr=7.99000e-06;faf95_joint_amr=0.00000;faf95_joint_eas=0.00000;faf95_joint_nfe=0.00000;faf95_joint_sas=0.00000;faf99_joint=8.20000e-07;faf99_joint_afr=2.99000e-06;faf99_joint_amr=0.00000;faf99_joint_eas=0.00000;faf99_joint_nfe=0.00000;faf99_joint_sas=0.00000;fafmax_faf95_max_joint=7.99000e-06;fafmax_faf95_max_gen_anc_joint=afr;fafmax_faf99_max_joint=2.99000e-06;fafmax_faf99_max_gen_anc_joint=afr;fafmax_data_type_joint=genomes;age_hist_het_bin_freq=0|0|1|1|0|0|0|0|0|0;age_hist_het_n_smaller=0;age_hist_het_n_larger=0;age_hist_hom_bin_freq=0|0|0|0|0|0|0|0|0|0;age_hist_hom_n_smaller=0;age_hist_hom_n_larger=0;FS=2.37367;MQ=49.0312;MQRankSum=-1.25400;QUALapprox=1760;QD=4.32432;ReadPosRankSum=-0.509000;SOR=0.950000;VarDP=407;AS_FS=2.37367;AS_MQ=49.0312;AS_MQRankSum=-1.25400;AS_pab_max=1.11230e-05;AS_QUALapprox=1760;AS_QD=4.32432;AS_ReadPosRankSum=-0.509000;AS_SB_TABLE=171,164|41,31;AS_SOR=0.949839;AS_VarDP=407;inbreeding_coeff=-2.62626e-05;AS_culprit=AS_QD;AS_VQSLOD=-2.15890;negative_train_site;allele_type=snv;n_alt_alleles=1;variant_type=snv;segdup;gq_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|2;gq_hist_all_bin_freq=0|0|0|0|475|319|375|355|461|530|583|689|1423|814|992|1015|1192|1340|1449|64142;dp_hist_alt_bin_freq=0|0|0|0|0|0|0|0|0|0|0|0|1|0|0|0|0|0|1|0;dp_hist_alt_n_larger=0;dp_hist_all_bin_freq=0|0|4|16|59|176|732|1896|2019|1864|2486|3271|4109|4922|5341|5731|5564|5478|5394|5925;dp_hist_all_n_larger=21167;ab_hist_alt_bin_freq=0|0|0|0|2|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;cadd_raw_score=0.289799;cadd_phred=4.17000;pangolin_largest_ds=0.00000;phylop=0.258000;VRS_Allele_IDs=ga4gh:VA.r54fNO45KZ5K7ugJ4orP7aaKfvCqdDCG,ga4gh:VA.9QBimEQMWMbgcJJKF2oE4derPs05JfN_;VRS_Starts=10000002,10000002;VRS_Ends=10000003,10000003;VRS_States=A,T;vep=T|upstream_gene_variant|MODIFIER||ENSG00000270533|Transcript|ENST00000604687|transcribed_processed_pseudogene||||||||||1|1970|-1||SNV|||YES||||||||Ensembl|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER||ENSG00000270533|Transcript|ENST00000625012|processed_transcript||3/10|ENST00000625012.1:n.449-1955T>A|||||||1||-1||SNV||||||1|||||Ensembl|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_001754952.1|lncRNA||5/6|XR_001754952.1:n.909-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_001754953.1|lncRNA||4/5|XR_001754953.1:n.788-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_001754954.1|lncRNA||4/5|XR_001754954.1:n.841-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_002958652.1|lncRNA||4/5|XR_002958652.1:n.797-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951164.2|lncRNA||7/8|XR_951164.2:n.1140-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951170.2|lncRNA||5/6|XR_951170.2:n.867-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951171.2|lncRNA||4/5|XR_951171.2:n.696-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951172.1|lncRNA||4/5|XR_951172.1:n.490-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951173.3|lncRNA||1/2|XR_951173.3:n.253-1955T>A|||||||1||-1||SNV|EntrezGene||||||||||RefSeq|||||||||||||||,T|intron_variant&non_coding_transcript_variant|MODIFIER|LOC105372733|105372733|Transcript|XR_951174.3|lncRNA||2/3|XR_951174.3:n.377-1955T>A|||||||1||-1||SNV|EntrezGene||YES||||||||RefSeq|||||||||||||||'"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "last_field"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3dcaaa8",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
