{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Introduction\n",
    "There are different types of possible splicing sites including (1) 5’ and 3’ splicing cites. These are signals marking the beginning and end of introns. Mutation in these regions can disrupt the splicing process. \n",
    "(2) Branch points. (3). Splicing regulatory elements, i.e. exonic/intronic splicing enhancers/silencers. Enhancers are sequences that promote splicing and are often found in exons and introns. Silencers are sequences that inhibit splicing. \n",
    "\n",
    "Single nucleotide variants (SNVs) are the most commonly observed genetic variants in the human genome. These SNVs represent a substantial portion of genetic variation and have the potential to influence splicing. \n",
    "\n",
    "We collect these variants from the following sources (1) HGMD\n",
    "which contains more than 13 000 mutations with consequences for mRNA splicing. (2) SpliceDisease database, which collects and curates experimentally supported data of RNA splicing mutations and disease. (3) Database for Aberrant Splice Sites (DBASS), which contains 577 and 307 records of mutation-induced and disease-causing aberrant 5’ and 3’ splice sites.(4) Somatic Mutations in Cancer (COSMIC) database. The data can be downloaded from websites\n",
    "https://hgdownload.soe.ucsc.edu/downloads.\n",
    "https://cancer.sanger.ac.uk/cosmic/download.\n",
    "\n",
    "For this presentation we take the existing composite data from the source data of paper.\n",
    "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4267638/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Source</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Position</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Group</th>\n",
       "      <th>Type</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Splice site</th>\n",
       "      <th>Distance</th>\n",
       "      <th>...</th>\n",
       "      <th>GENSCAN_ref</th>\n",
       "      <th>GENSCAN_alt</th>\n",
       "      <th>NetGene2_ref</th>\n",
       "      <th>NetGene2_alt</th>\n",
       "      <th>SplicePredictor_ref</th>\n",
       "      <th>SplicePredictor_alt</th>\n",
       "      <th>phyloP46way_placental</th>\n",
       "      <th>phyloP46way_primate</th>\n",
       "      <th>CADD_raw</th>\n",
       "      <th>CADD_phred</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000G</td>\n",
       "      <td>1</td>\n",
       "      <td>889159</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Intronic</td>\n",
       "      <td>-</td>\n",
       "      <td>5'</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.31</td>\n",
       "      <td>-0.37</td>\n",
       "      <td>0.52</td>\n",
       "      <td>6.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000G</td>\n",
       "      <td>1</td>\n",
       "      <td>982941</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Intronic</td>\n",
       "      <td>+</td>\n",
       "      <td>3'</td>\n",
       "      <td>-12</td>\n",
       "      <td>...</td>\n",
       "      <td>0.970</td>\n",
       "      <td>0.966</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.900</td>\n",
       "      <td>0.859</td>\n",
       "      <td>-0.30</td>\n",
       "      <td>-1.53</td>\n",
       "      <td>-1.29</td>\n",
       "      <td>0.04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000G</td>\n",
       "      <td>1</td>\n",
       "      <td>1891477</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Intronic</td>\n",
       "      <td>-</td>\n",
       "      <td>3'</td>\n",
       "      <td>-8</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.46</td>\n",
       "      <td>0.45</td>\n",
       "      <td>1.11</td>\n",
       "      <td>9.52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000G</td>\n",
       "      <td>1</td>\n",
       "      <td>2528133</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Intronic</td>\n",
       "      <td>-</td>\n",
       "      <td>3'</td>\n",
       "      <td>-5</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-2.62</td>\n",
       "      <td>-4.05</td>\n",
       "      <td>-1.02</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000G</td>\n",
       "      <td>1</td>\n",
       "      <td>2541269</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Exonic</td>\n",
       "      <td>-</td>\n",
       "      <td>3'</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.96</td>\n",
       "      <td>-0.23</td>\n",
       "      <td>1.31</td>\n",
       "      <td>10.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1000G</td>\n",
       "      <td>1</td>\n",
       "      <td>3395174</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Exonic</td>\n",
       "      <td>+</td>\n",
       "      <td>5'</td>\n",
       "      <td>-3</td>\n",
       "      <td>...</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.892</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.996</td>\n",
       "      <td>0.996</td>\n",
       "      <td>-2.38</td>\n",
       "      <td>-2.07</td>\n",
       "      <td>-1.76</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1000G</td>\n",
       "      <td>1</td>\n",
       "      <td>3761465</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Intronic</td>\n",
       "      <td>-</td>\n",
       "      <td>5'</td>\n",
       "      <td>6</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.38</td>\n",
       "      <td>-0.31</td>\n",
       "      <td>0.55</td>\n",
       "      <td>7.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1000G</td>\n",
       "      <td>1</td>\n",
       "      <td>3806643</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Intronic</td>\n",
       "      <td>-</td>\n",
       "      <td>3'</td>\n",
       "      <td>-6</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>-0.69</td>\n",
       "      <td>-0.36</td>\n",
       "      <td>-0.16</td>\n",
       "      <td>3.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1000G</td>\n",
       "      <td>1</td>\n",
       "      <td>6204222</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Intronic</td>\n",
       "      <td>-</td>\n",
       "      <td>3'</td>\n",
       "      <td>-7</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>2.39</td>\n",
       "      <td>0.46</td>\n",
       "      <td>1.15</td>\n",
       "      <td>9.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1000G</td>\n",
       "      <td>1</td>\n",
       "      <td>7737651</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Intronic</td>\n",
       "      <td>+</td>\n",
       "      <td>3'</td>\n",
       "      <td>-8</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.996</td>\n",
       "      <td>0.996</td>\n",
       "      <td>-0.30</td>\n",
       "      <td>-0.26</td>\n",
       "      <td>1.05</td>\n",
       "      <td>9.30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Source Chr  Position Ref Alt     Group      Type Strand Splice site  \\\n",
       "0  1000G   1    889159   A   C  Negative  Intronic      -          5'   \n",
       "1  1000G   1    982941   T   C  Negative  Intronic      +          3'   \n",
       "2  1000G   1   1891477   A   G  Negative  Intronic      -          3'   \n",
       "3  1000G   1   2528133   T   C  Negative  Intronic      -          3'   \n",
       "4  1000G   1   2541269   A   G  Negative    Exonic      -          3'   \n",
       "5  1000G   1   3395174   C   A  Negative    Exonic      +          5'   \n",
       "6  1000G   1   3761465   T   A  Negative  Intronic      -          5'   \n",
       "7  1000G   1   3806643   T   A  Negative  Intronic      -          3'   \n",
       "8  1000G   1   6204222   C   G  Negative  Intronic      -          3'   \n",
       "9  1000G   1   7737651   G   A  Negative  Intronic      +          3'   \n",
       "\n",
       "   Distance  ...  GENSCAN_ref  GENSCAN_alt  NetGene2_ref  NetGene2_alt  \\\n",
       "0         3  ...        0.000        0.000           0.0           0.0   \n",
       "1       -12  ...        0.970        0.966           0.0           0.0   \n",
       "2        -8  ...        0.000        0.000           0.0           0.0   \n",
       "3        -5  ...        0.000        0.000           0.0           0.0   \n",
       "4         2  ...        0.000        0.000           0.0           0.0   \n",
       "5        -3  ...        0.902        0.892           1.0           1.0   \n",
       "6         6  ...        0.000        0.000           0.0           0.0   \n",
       "7        -6  ...        0.000        0.000           0.0           0.0   \n",
       "8        -7  ...        0.000        0.000           0.0           0.0   \n",
       "9        -8  ...        0.000        0.000           0.0           0.0   \n",
       "\n",
       "   SplicePredictor_ref  SplicePredictor_alt  phyloP46way_placental  \\\n",
       "0                0.000                0.000                  -0.31   \n",
       "1                0.900                0.859                  -0.30   \n",
       "2                0.000                0.000                  -0.46   \n",
       "3                0.000                0.000                  -2.62   \n",
       "4                0.000                0.000                  -0.96   \n",
       "5                0.996                0.996                  -2.38   \n",
       "6                0.000                0.000                  -0.38   \n",
       "7                0.000                0.000                  -0.69   \n",
       "8                0.000                0.000                   2.39   \n",
       "9                0.996                0.996                  -0.30   \n",
       "\n",
       "   phyloP46way_primate  CADD_raw  CADD_phred  \n",
       "0                -0.37      0.52        6.80  \n",
       "1                -1.53     -1.29        0.04  \n",
       "2                 0.45      1.11        9.52  \n",
       "3                -4.05     -1.02        0.23  \n",
       "4                -0.23      1.31       10.28  \n",
       "5                -2.07     -1.76        0.01  \n",
       "6                -0.31      0.55        7.00  \n",
       "7                -0.36     -0.16        3.20  \n",
       "8                 0.46      1.15        9.70  \n",
       "9                -0.26      1.05        9.30  \n",
       "\n",
       "[10 rows x 30 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# here we download the existing composite data from the source data of paper \n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "composite = pd.read_excel(\"paper\\\\Supplementary_Table_S1-S6-In silico prediction of splice-altering single nucleotide variants in the human genome.xlsx\", sheet_name=\"Table S2\", header=1)\n",
    "\n",
    "composite.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Position</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>No. of times observed</th>\n",
       "      <th>Functional region</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ada_score</th>\n",
       "      <th>rf_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>2489782</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>nonsynonymous</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>0.999894</td>\n",
       "      <td>0.994</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>2491418</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>0.999994</td>\n",
       "      <td>0.760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>3313158</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.999967</td>\n",
       "      <td>0.898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>3319354</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.999972</td>\n",
       "      <td>0.926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>3334391</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.999939</td>\n",
       "      <td>0.956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>3334561</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>nonsynonymous</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.999060</td>\n",
       "      <td>0.952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>3342312</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>nonsynonymous</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>0.993861</td>\n",
       "      <td>0.864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>6246877</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>RPL22</td>\n",
       "      <td>0.999988</td>\n",
       "      <td>0.926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>7527961</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>nonsynonymous</td>\n",
       "      <td>CAMTA1</td>\n",
       "      <td>0.999951</td>\n",
       "      <td>0.996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>7723412</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>CAMTA1</td>\n",
       "      <td>0.999952</td>\n",
       "      <td>0.958</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr  Position Ref Alt  No. of times observed Functional region      Gene  \\\n",
       "0   1   2489782   G   T                      1     nonsynonymous  TNFRSF14   \n",
       "1   1   2491418   G   A                      1          splicing  TNFRSF14   \n",
       "2   1   3313158   G   A                      1          splicing    PRDM16   \n",
       "3   1   3319354   G   T                      1          splicing    PRDM16   \n",
       "4   1   3334391   G   T                      1          splicing    PRDM16   \n",
       "5   1   3334561   G   A                      1     nonsynonymous    PRDM16   \n",
       "6   1   3342312   C   T                      1     nonsynonymous    PRDM16   \n",
       "7   1   6246877   C   A                      1          splicing     RPL22   \n",
       "8   1   7527961   G   T                      1     nonsynonymous    CAMTA1   \n",
       "9   1   7723412   G   T                      1          splicing    CAMTA1   \n",
       "\n",
       "   ada_score  rf_score  \n",
       "0   0.999894     0.994  \n",
       "1   0.999994     0.760  \n",
       "2   0.999967     0.898  \n",
       "3   0.999972     0.926  \n",
       "4   0.999939     0.956  \n",
       "5   0.999060     0.952  \n",
       "6   0.993861     0.864  \n",
       "7   0.999988     0.926  \n",
       "8   0.999951     0.996  \n",
       "9   0.999952     0.958  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# this is COSMIC data only\n",
    "\n",
    "cosmic = pd.read_excel(\"paper\\\\Supplementary_Table_S1-S6-In silico prediction of splice-altering single nucleotide variants in the human genome.xlsx\", sheet_name=\"Table S6\", header=1)\n",
    "\n",
    "cosmic.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Position</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>No. of times observed</th>\n",
       "      <th>Functional region</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ada_score</th>\n",
       "      <th>rf_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1646</th>\n",
       "      <td>17</td>\n",
       "      <td>41201212</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999950</td>\n",
       "      <td>0.950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1647</th>\n",
       "      <td>17</td>\n",
       "      <td>41203135</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999989</td>\n",
       "      <td>0.934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1648</th>\n",
       "      <td>17</td>\n",
       "      <td>41209068</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999987</td>\n",
       "      <td>0.930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1649</th>\n",
       "      <td>17</td>\n",
       "      <td>41209068</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999987</td>\n",
       "      <td>0.930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1650</th>\n",
       "      <td>17</td>\n",
       "      <td>41215391</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>0.928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1651</th>\n",
       "      <td>17</td>\n",
       "      <td>41223256</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999978</td>\n",
       "      <td>0.758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1652</th>\n",
       "      <td>17</td>\n",
       "      <td>41256137</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>6</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999987</td>\n",
       "      <td>0.944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1653</th>\n",
       "      <td>17</td>\n",
       "      <td>41256137</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>3</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999987</td>\n",
       "      <td>0.946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1654</th>\n",
       "      <td>17</td>\n",
       "      <td>41267742</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999990</td>\n",
       "      <td>0.938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1655</th>\n",
       "      <td>17</td>\n",
       "      <td>41276033</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999978</td>\n",
       "      <td>0.944</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr  Position Ref Alt  No. of times observed Functional region   Gene  \\\n",
       "1646  17  41201212   C   A                      1          splicing  BRCA1   \n",
       "1647  17  41203135   C   A                      2          splicing  BRCA1   \n",
       "1648  17  41209068   C   A                      1          splicing  BRCA1   \n",
       "1649  17  41209068   C   T                      1          splicing  BRCA1   \n",
       "1650  17  41215391   C   T                      1          splicing  BRCA1   \n",
       "1651  17  41223256   C   T                      1          splicing  BRCA1   \n",
       "1652  17  41256137   A   C                      6          splicing  BRCA1   \n",
       "1653  17  41256137   A   T                      3          splicing  BRCA1   \n",
       "1654  17  41267742   C   A                      1          splicing  BRCA1   \n",
       "1655  17  41276033   C   T                      1          splicing  BRCA1   \n",
       "\n",
       "      ada_score  rf_score  \n",
       "1646   0.999950     0.950  \n",
       "1647   0.999989     0.934  \n",
       "1648   0.999987     0.930  \n",
       "1649   0.999987     0.930  \n",
       "1650   0.999990     0.928  \n",
       "1651   0.999978     0.758  \n",
       "1652   0.999987     0.944  \n",
       "1653   0.999987     0.946  \n",
       "1654   0.999990     0.938  \n",
       "1655   0.999978     0.944  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cosmic_gene = cosmic[cosmic['Gene'] == 'BRCA1']\n",
    "\n",
    "cosmic_gene"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge with gnomAD population variants dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    Chromosome  Position         rsIDs         Transcript\n",
      "0           17  41197628   rs754327493  ENST00000309486.4\n",
      "1           17  41197632   rs539044217  ENST00000309486.4\n",
      "2           17  41197635   rs778988967  ENST00000309486.4\n",
      "3           17  41197637   rs137892861  ENST00000309486.4\n",
      "4           17  41197637   rs137892861  ENST00000309486.4\n",
      "5           17  41197652   rs758539310  ENST00000471181.2\n",
      "6           17  41197656   rs777938968  ENST00000471181.2\n",
      "7           17  41197659     rs3092995  ENST00000471181.2\n",
      "8           17  41197659     rs3092995  ENST00000471181.2\n",
      "9           17  41197664   rs776323505  ENST00000471181.2\n",
      "10          17  41197669   rs745752144  ENST00000471181.2\n",
      "11          17  41197672  rs1305255318  ENST00000471181.2\n",
      "12          17  41197673  rs1346384762  ENST00000471181.2\n",
      "13          17  41197674  rs1232678023  ENST00000471181.2\n",
      "14          17  41197675   rs375042815  ENST00000471181.2\n",
      "15          17  41197682   rs762552027  ENST00000471181.2\n",
      "16          17  41197698  rs1224760267  ENST00000471181.2\n",
      "17          17  41197700   rs763740623  ENST00000471181.2\n",
      "18          17  41197701   rs774127304  ENST00000471181.2\n",
      "19          17  41197702    rs80357183  ENST00000471181.2\n"
     ]
    }
   ],
   "source": [
    "import csv\n",
    "\n",
    "filename = 'gnomAD_v2.1.1_ENSG00000012048_2023_10_12_16_02_58.csv'\n",
    "\n",
    "# Read the CSV file into a DataFrame\n",
    "gnom = pd.read_csv(filename)\n",
    "\n",
    "# Display the first few rows of the DataFrame\n",
    "print(gnom[['Chromosome', 'Position', 'rsIDs', 'Transcript']].head(20))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Position</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>No. of times observed</th>\n",
       "      <th>Functional region</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ada_score</th>\n",
       "      <th>rf_score</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>...</th>\n",
       "      <th>Homozygote Count European (non-Finnish)</th>\n",
       "      <th>Hemizygote Count European (non-Finnish)</th>\n",
       "      <th>Allele Count Other</th>\n",
       "      <th>Allele Number Other</th>\n",
       "      <th>Homozygote Count Other</th>\n",
       "      <th>Hemizygote Count Other</th>\n",
       "      <th>Allele Count South Asian</th>\n",
       "      <th>Allele Number South Asian</th>\n",
       "      <th>Homozygote Count South Asian</th>\n",
       "      <th>Hemizygote Count South Asian</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>41203135</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999989</td>\n",
       "      <td>0.934</td>\n",
       "      <td>17</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6126</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30602</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr  Position Ref Alt  No. of times observed Functional region   Gene  \\\n",
       "0  17  41203135   C   A                      2          splicing  BRCA1   \n",
       "\n",
       "   ada_score  rf_score  Chromosome  ...  \\\n",
       "0   0.999989     0.934          17  ...   \n",
       "\n",
       "  Homozygote Count European (non-Finnish)  \\\n",
       "0                                       0   \n",
       "\n",
       "  Hemizygote Count European (non-Finnish) Allele Count Other  \\\n",
       "0                                       0                  0   \n",
       "\n",
       "  Allele Number Other Homozygote Count Other Hemizygote Count Other  \\\n",
       "0                6126                      0                      0   \n",
       "\n",
       "  Allele Count South Asian Allele Number South Asian  \\\n",
       "0                        0                     30602   \n",
       "\n",
       "  Homozygote Count South Asian Hemizygote Count South Asian  \n",
       "0                            0                            0  \n",
       "\n",
       "[1 rows x 61 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = pd.merge(cosmic, gnom, on='Position', how='inner')\n",
    "\n",
    "merged_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### One challenge at this point:\n",
    "\n",
    "The first is one can see that there in only one entry after merging. Note that the Position/location between \n",
    "two dataset need not to be exactly the same. So we need turn the position from one dataset into a range, and use the range to cover more rows. For this we use Ensembl API to get the range of the variant position."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Solution: using Ensembl API to get ranges for each variant location"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene Symbol: BRCA1\n",
      "Gene Description: BRCA1 DNA repair associated [Source:HGNC Symbol;Acc:HGNC:1100]\n"
     ]
    }
   ],
   "source": [
    "gene = 'PCSK9'\n",
    "\n",
    "gene = 'ARID1A'\n",
    "\n",
    "gene = 'BRCA1'\n",
    "\n",
    "#homo_sapiens\n",
    "\n",
    "api_url = f\"https://rest.ensembl.org/lookup/symbol/human/{gene}?expand=1\"\n",
    "\n",
    "# Send a GET request to the Ensembl API.\n",
    "response = requests.get(api_url, headers={\"Content-Type\": \"application/json\"})\n",
    "\n",
    "# Check for a successful response.\n",
    "if response.status_code == 200:\n",
    "    gene_info = response.json()\n",
    "    print(\"Gene Symbol:\", gene_info[\"display_name\"])\n",
    "    print(\"Gene Description:\", gene_info[\"description\"])\n",
    "else:\n",
    "    print(\"Failed to retrieve gene information.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>object_type</th>\n",
       "      <th>source</th>\n",
       "      <th>strand</th>\n",
       "      <th>seq_region_name</th>\n",
       "      <th>db_type</th>\n",
       "      <th>Parent</th>\n",
       "      <th>is_canonical</th>\n",
       "      <th>start</th>\n",
       "      <th>Translation</th>\n",
       "      <th>display_name</th>\n",
       "      <th>biotype</th>\n",
       "      <th>logic_name</th>\n",
       "      <th>end</th>\n",
       "      <th>id</th>\n",
       "      <th>species</th>\n",
       "      <th>version</th>\n",
       "      <th>Exon</th>\n",
       "      <th>assembly_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcript</td>\n",
       "      <td>havana</td>\n",
       "      <td>-1</td>\n",
       "      <td>17</td>\n",
       "      <td>core</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>0</td>\n",
       "      <td>43044295</td>\n",
       "      <td>{'length': 1567, 'object_type': 'Translation',...</td>\n",
       "      <td>BRCA1-222</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>havana_homo_sapiens</td>\n",
       "      <td>43125300</td>\n",
       "      <td>ENST00000497488</td>\n",
       "      <td>human</td>\n",
       "      <td>2</td>\n",
       "      <td>[{'id': 'ENSE00001867485', 'strand': -1, 'seq_...</td>\n",
       "      <td>GRCh38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Transcript</td>\n",
       "      <td>havana</td>\n",
       "      <td>-1</td>\n",
       "      <td>17</td>\n",
       "      <td>core</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>0</td>\n",
       "      <td>43044295</td>\n",
       "      <td>{'object_type': 'Translation', 'length': 1837,...</td>\n",
       "      <td>BRCA1-216</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>havana_homo_sapiens</td>\n",
       "      <td>43125321</td>\n",
       "      <td>ENST00000489037</td>\n",
       "      <td>human</td>\n",
       "      <td>2</td>\n",
       "      <td>[{'object_type': 'Exon', 'end': 43125321, 'str...</td>\n",
       "      <td>GRCh38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Transcript</td>\n",
       "      <td>havana</td>\n",
       "      <td>-1</td>\n",
       "      <td>17</td>\n",
       "      <td>core</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>0</td>\n",
       "      <td>43044295</td>\n",
       "      <td>{'version': 2, 'start': 43045678, 'end': 43124...</td>\n",
       "      <td>BRCA1-214</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>havana_homo_sapiens</td>\n",
       "      <td>43125359</td>\n",
       "      <td>ENST00000478531</td>\n",
       "      <td>human</td>\n",
       "      <td>6</td>\n",
       "      <td>[{'start': 43125277, 'version': 1, 'assembly_n...</td>\n",
       "      <td>GRCh38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Transcript</td>\n",
       "      <td>ensembl_havana</td>\n",
       "      <td>-1</td>\n",
       "      <td>17</td>\n",
       "      <td>core</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>1</td>\n",
       "      <td>43044295</td>\n",
       "      <td>{'start': 43045678, 'version': 3, 'end': 43124...</td>\n",
       "      <td>BRCA1-203</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ensembl_havana_transcript_homo_sapiens</td>\n",
       "      <td>43125364</td>\n",
       "      <td>ENST00000357654</td>\n",
       "      <td>human</td>\n",
       "      <td>9</td>\n",
       "      <td>[{'object_type': 'Exon', 'species': 'human', '...</td>\n",
       "      <td>GRCh38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Transcript</td>\n",
       "      <td>havana</td>\n",
       "      <td>-1</td>\n",
       "      <td>17</td>\n",
       "      <td>core</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>0</td>\n",
       "      <td>43044295</td>\n",
       "      <td>{'start': 43045678, 'version': 2, 'end': 43124...</td>\n",
       "      <td>BRCA1-211</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>havana_homo_sapiens</td>\n",
       "      <td>43125364</td>\n",
       "      <td>ENST00000473961</td>\n",
       "      <td>human</td>\n",
       "      <td>6</td>\n",
       "      <td>[{'end': 43125364, 'strand': -1, 'id': 'ENSE00...</td>\n",
       "      <td>GRCh38</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  object_type          source  strand seq_region_name db_type  \\\n",
       "0  Transcript          havana      -1              17    core   \n",
       "1  Transcript          havana      -1              17    core   \n",
       "2  Transcript          havana      -1              17    core   \n",
       "3  Transcript  ensembl_havana      -1              17    core   \n",
       "4  Transcript          havana      -1              17    core   \n",
       "\n",
       "            Parent  is_canonical     start  \\\n",
       "0  ENSG00000012048             0  43044295   \n",
       "1  ENSG00000012048             0  43044295   \n",
       "2  ENSG00000012048             0  43044295   \n",
       "3  ENSG00000012048             1  43044295   \n",
       "4  ENSG00000012048             0  43044295   \n",
       "\n",
       "                                         Translation display_name  \\\n",
       "0  {'length': 1567, 'object_type': 'Translation',...    BRCA1-222   \n",
       "1  {'object_type': 'Translation', 'length': 1837,...    BRCA1-216   \n",
       "2  {'version': 2, 'start': 43045678, 'end': 43124...    BRCA1-214   \n",
       "3  {'start': 43045678, 'version': 3, 'end': 43124...    BRCA1-203   \n",
       "4  {'start': 43045678, 'version': 2, 'end': 43124...    BRCA1-211   \n",
       "\n",
       "          biotype                              logic_name       end  \\\n",
       "0  protein_coding                     havana_homo_sapiens  43125300   \n",
       "1  protein_coding                     havana_homo_sapiens  43125321   \n",
       "2  protein_coding                     havana_homo_sapiens  43125359   \n",
       "3  protein_coding  ensembl_havana_transcript_homo_sapiens  43125364   \n",
       "4  protein_coding                     havana_homo_sapiens  43125364   \n",
       "\n",
       "                id species  version  \\\n",
       "0  ENST00000497488   human        2   \n",
       "1  ENST00000489037   human        2   \n",
       "2  ENST00000478531   human        6   \n",
       "3  ENST00000357654   human        9   \n",
       "4  ENST00000473961   human        6   \n",
       "\n",
       "                                                Exon assembly_name  \n",
       "0  [{'id': 'ENSE00001867485', 'strand': -1, 'seq_...        GRCh38  \n",
       "1  [{'object_type': 'Exon', 'end': 43125321, 'str...        GRCh38  \n",
       "2  [{'start': 43125277, 'version': 1, 'assembly_n...        GRCh38  \n",
       "3  [{'object_type': 'Exon', 'species': 'human', '...        GRCh38  \n",
       "4  [{'end': 43125364, 'strand': -1, 'id': 'ENSE00...        GRCh38  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl = pd.DataFrame(gene_info['Transcript'])\n",
    "\n",
    "ensembl.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Position</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>No. of times observed</th>\n",
       "      <th>Functional region</th>\n",
       "      <th>Gene</th>\n",
       "      <th>ada_score</th>\n",
       "      <th>rf_score</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>17</td>\n",
       "      <td>41276033</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>splicing</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>0.999978</td>\n",
       "      <td>0.944</td>\n",
       "      <td>41274360</td>\n",
       "      <td>41277387</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chr  Position Ref Alt  No. of times observed Functional region   Gene  \\\n",
       "9   17  41276033   C   T                      1          splicing  BRCA1   \n",
       "\n",
       "   ada_score  rf_score     start       end  \n",
       "9   0.999978     0.944  41274360  41277387  "
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pyliftover import LiftOver\n",
    "\n",
    "lo = LiftOver('Hg38Tohg19.over.chain.gz')\n",
    "\n",
    "# List to store merged rows\n",
    "merged_rows = []\n",
    "\n",
    "# Loop through each row in df1 and df2\n",
    "for index1, row1 in cosmic_gene.iterrows():\n",
    "    \n",
    "    min_start, max_end = row1['Position'], row1['Position']\n",
    "    \n",
    "    for index2, row2 in ensembl.iterrows():\n",
    "        \n",
    "        #print(row2['start'], row2['end'])\n",
    "        \n",
    "        start = row2['start']\n",
    "        end = row2['end']\n",
    "        chrom = 'chr' + str(row1['Chr'])\n",
    "\n",
    "        converted = lo.convert_coordinate(chrom, start)\n",
    "        if converted:\n",
    "            new_chrom, new_start, _, _ = converted[0]\n",
    "        else:\n",
    "            continue\n",
    "            \n",
    "        converted = lo.convert_coordinate(chrom, end)\n",
    "        if converted:\n",
    "            new_chrom, new_end, _, _ = converted[0]\n",
    "        else:\n",
    "            continue\n",
    "            \n",
    "        min_start = min(min_start, new_start)\n",
    "        \n",
    "        max_end = max(max_end, new_end)\n",
    "        \n",
    "    if min_start <= row1['Position'] <= max_end:\n",
    "        # Combine the rows\n",
    "        combined_row = row1.to_dict()\n",
    "        combined_row.update({'start': new_start,  'end': new_end})\n",
    "        merged_rows.append(combined_row)\n",
    "\n",
    "# Convert merged rows to DataFrame\n",
    "cosmic_gene_ensembl = pd.DataFrame(merged_rows)\n",
    "\n",
    "# remove duplicated ranges\n",
    "cosmic_gene_ensembl = cosmic_gene_ensembl.drop_duplicates(subset=['start', 'end'], keep='last')\n",
    "\n",
    "cosmic_gene_ensembl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Mapping ranges to gnomAD variants data for each gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    Chromosome  Position         rsIDs\n",
      "0           17  41275983   rs180905862\n",
      "1           17  41275986   rs200513210\n",
      "2           17  41275996   rs768583394\n",
      "3           17  41276007  rs1213255171\n",
      "4           17  41276011  rs1447426729\n",
      "..         ...       ...           ...\n",
      "91          17  41277260   rs904265019\n",
      "92          17  41277261  rs1172227698\n",
      "93          17  41277262   rs746485760\n",
      "94          17  41277275   rs770368390\n",
      "95          17  41277277   rs273898672\n",
      "\n",
      "[96 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "# List to store merged rows\n",
    "merged_rows = []\n",
    "\n",
    "# Loop through each row in df1 and df2\n",
    "for index1, row1 in cosmic_gene_ensembl.iterrows():\n",
    "    for index2, row2 in gnom.iterrows():\n",
    "        if row1['start'] <= row2['Position'] <= row1['end']:\n",
    "            # Combine the rows\n",
    "            combined_row = row1.to_dict()\n",
    "            combined_row.update(row2.to_dict())\n",
    "            merged_rows.append(combined_row)\n",
    "\n",
    "# Convert merged rows to DataFrame\n",
    "merged_df = pd.DataFrame(merged_rows)\n",
    "\n",
    "if len(merged_df) > 0:\n",
    "    \n",
    "    merged_df[['Position', 'start', 'end']]\n",
    "\n",
    "    print(merged_df[['Chromosome', 'Position', 'rsIDs']])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
