{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1 55505100 55505250\n",
      "chr1 55505385 55505535\n",
      "chr1 55504905 55505055\n",
      "chr1 55586360 55586510\n",
      "chr1 55411220 55411370\n",
      "chr1 55525845 55525995\n",
      "chr1 55534780 55534930\n",
      "chr1 55509965 55510115\n",
      "chr1 55506205 55506355\n",
      "chr1 55504305 55504455\n"
     ]
    }
   ],
   "source": [
    "bed_file = 'connect_ENSG00000169174.bed'\n",
    "\n",
    "with open(bed_file, 'r') as f:\n",
    "    for line in f:\n",
    "        if len(line.strip().split()) >= 3:\n",
    "            chrom, start, end = line.strip().split()[:3]\n",
    "            print(chrom, start, end)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55505100</td>\n",
       "      <td>55505250</td>\n",
       "      <td>120.199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "      <td>202.571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55504905</td>\n",
       "      <td>55505055</td>\n",
       "      <td>103.534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55586360</td>\n",
       "      <td>55586510</td>\n",
       "      <td>79.1868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55411220</td>\n",
       "      <td>55411370</td>\n",
       "      <td>122.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55525845</td>\n",
       "      <td>55525995</td>\n",
       "      <td>73.5834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55534780</td>\n",
       "      <td>55534930</td>\n",
       "      <td>104.872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55509965</td>\n",
       "      <td>55510115</td>\n",
       "      <td>134.802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55506205</td>\n",
       "      <td>55506355</td>\n",
       "      <td>98.7568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55504305</td>\n",
       "      <td>55504455</td>\n",
       "      <td>112.666</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome     start       end    value\n",
       "0       chr1  55505100  55505250  120.199\n",
       "1       chr1  55505385  55505535  202.571\n",
       "2       chr1  55504905  55505055  103.534\n",
       "3       chr1  55586360  55586510  79.1868\n",
       "4       chr1  55411220  55411370   122.74\n",
       "5       chr1  55525845  55525995  73.5834\n",
       "6       chr1  55534780  55534930  104.872\n",
       "7       chr1  55509965  55510115  134.802\n",
       "8       chr1  55506205  55506355  98.7568\n",
       "9       chr1  55504305  55504455  112.666"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "bed_file = 'connect_ENSG00000169174.bed'\n",
    "\n",
    "# Open the file\n",
    "with open(bed_file, 'r') as f:\n",
    "    # Read the header (column names)\n",
    "    #header = f.readline().strip().split(',')\n",
    "    header =[\"chromosome\",\"start\",\"end\", \"value\"]\n",
    "    \n",
    "    # Initialize a dictionary with empty lists for each column\n",
    "    data = {col: [] for col in header}\n",
    "    \n",
    "    # Iterate over each line in the file\n",
    "    for line in f:\n",
    "        # Split the line into values\n",
    "        values = line.strip().split()\n",
    "\n",
    "        if len(values) >= 3:\n",
    "            # Append values to the corresponding lists in the dictionary\n",
    "            for i, col in enumerate(header):\n",
    "                data[col].append(values[i])\n",
    "\n",
    "# Convert the dictionary to a DataFrame\n",
    "rf = pd.DataFrame(data)\n",
    "\n",
    "\n",
    "rf['start'] = rf['start'].astype(int)\n",
    "\n",
    "rf['end'] = rf['end'].astype(int)\n",
    "\n",
    "rf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Chromosome  Position         rsIDs Reference Alternate          Source  \\\n",
      "0           1  55505447    rs45448095         C         T  gnomAD Genomes   \n",
      "1           1  55505452   rs576575389         T         G  gnomAD Genomes   \n",
      "2           1  55505457   rs545454601         C         T  gnomAD Genomes   \n",
      "3           1  55505457   rs545454601         C         G  gnomAD Genomes   \n",
      "4           1  55505462  rs1043549074         A         G   gnomAD Exomes   \n",
      "\n",
      "  Filters - exomes Filters - genomes         Transcript HGVS Consequence  ...  \\\n",
      "0              NaN              PASS  ENST00000302118.5         c.-64C>T  ...   \n",
      "1              NaN              PASS  ENST00000302118.5         c.-59T>G  ...   \n",
      "2              NaN              PASS  ENST00000302118.5         c.-54C>T  ...   \n",
      "3              NaN              PASS  ENST00000302118.5         c.-54C>G  ...   \n",
      "4             PASS               NaN  ENST00000302118.5         c.-49A>G  ...   \n",
      "\n",
      "  Homozygote Count European (non-Finnish)  \\\n",
      "0                                     107   \n",
      "1                                       0   \n",
      "2                                       0   \n",
      "3                                       0   \n",
      "4                                       0   \n",
      "\n",
      "  Hemizygote Count European (non-Finnish) Allele Count Other  \\\n",
      "0                                       0                108   \n",
      "1                                       0                  0   \n",
      "2                                       0                  0   \n",
      "3                                       0                  0   \n",
      "4                                       0                  0   \n",
      "\n",
      "  Allele Number Other  Homozygote Count Other Hemizygote Count Other  \\\n",
      "0                1086                       5                      0   \n",
      "1                1086                       0                      0   \n",
      "2                1088                       0                      0   \n",
      "3                1088                       0                      0   \n",
      "4                4680                       0                      0   \n",
      "\n",
      "   Allele Count South Asian  Allele Number South Asian  \\\n",
      "0                         0                          0   \n",
      "1                         0                          0   \n",
      "2                         0                          0   \n",
      "3                         0                          0   \n",
      "4                         0                      24070   \n",
      "\n",
      "   Homozygote Count South Asian  Hemizygote Count South Asian  \n",
      "0                             0                             0  \n",
      "1                             0                             0  \n",
      "2                             0                             0  \n",
      "3                             0                             0  \n",
      "4                             0                             0  \n",
      "\n",
      "[5 rows x 53 columns]\n"
     ]
    }
   ],
   "source": [
    "import csv\n",
    "\n",
    "filename = 'gnomAD_v2.1.1_ENSG00000169174_2023_09_10_17_58_40.csv'\n",
    "\n",
    "# Read the CSV file into a DataFrame\n",
    "gf = pd.read_csv(filename)\n",
    "\n",
    "# Display the first few rows of the DataFrame\n",
    "print(gf.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    Chromosome  Position         rsIDs Reference Alternate  \\\n",
      "0            1  55505447    rs45448095         C         T   \n",
      "1            1  55505452   rs576575389         T         G   \n",
      "2            1  55505457   rs545454601         C         T   \n",
      "3            1  55505457   rs545454601         C         G   \n",
      "4            1  55505462  rs1043549074         A         G   \n",
      "5            1  55505463  rs1193796116         C         T   \n",
      "6            1  55505464  rs1269395517         G         C   \n",
      "7            1  55505466  rs1365420733         C         T   \n",
      "8            1  55505469  rs1473950917         C         T   \n",
      "9            1  55505477   rs576535214         C         T   \n",
      "10           1  55505482  rs1389098623         C         T   \n",
      "11           1  55505483   rs376542717         C         G   \n",
      "12           1  55505485    rs28362202         G         A   \n",
      "13           1  55505488  rs1393600693         C         T   \n",
      "14           1  55505489  rs1387827816         A         C   \n",
      "15           1  55505499   rs182393980         C         A   \n",
      "16           1  55505500  rs1298619732         C         T   \n",
      "17           1  55505501  rs1341655436         C         T   \n",
      "18           1  55505503   rs886039837         T         C   \n",
      "19           1  55505507  rs1357021576         C         A   \n",
      "20           1  55505512  rs1209186979         T         C   \n",
      "21           1  55505515  rs1018576699         G         A   \n",
      "22           1  55505519  rs1465865735         C         T   \n",
      "23           1  55505520   rs186669805         G         A   \n",
      "\n",
      "                          Source Filters - exomes Filters - genomes  \\\n",
      "0                 gnomAD Genomes              NaN              PASS   \n",
      "1                 gnomAD Genomes              NaN              PASS   \n",
      "2                 gnomAD Genomes              NaN              PASS   \n",
      "3                 gnomAD Genomes              NaN              PASS   \n",
      "4                  gnomAD Exomes             PASS               NaN   \n",
      "5                  gnomAD Exomes             PASS               NaN   \n",
      "6   gnomAD Exomes,gnomAD Genomes             PASS              PASS   \n",
      "7                 gnomAD Genomes              NaN              PASS   \n",
      "8                 gnomAD Genomes              NaN              PASS   \n",
      "9   gnomAD Exomes,gnomAD Genomes             PASS              PASS   \n",
      "10                 gnomAD Exomes             PASS               NaN   \n",
      "11                gnomAD Genomes              NaN              PASS   \n",
      "12  gnomAD Exomes,gnomAD Genomes             PASS              PASS   \n",
      "13                 gnomAD Exomes             PASS               NaN   \n",
      "14  gnomAD Exomes,gnomAD Genomes             PASS              PASS   \n",
      "15                 gnomAD Exomes             PASS               NaN   \n",
      "16                 gnomAD Exomes             PASS               NaN   \n",
      "17                 gnomAD Exomes             PASS               NaN   \n",
      "18  gnomAD Exomes,gnomAD Genomes             PASS              PASS   \n",
      "19                 gnomAD Exomes             PASS               NaN   \n",
      "20                 gnomAD Exomes             PASS               NaN   \n",
      "21                 gnomAD Exomes             PASS               NaN   \n",
      "22                 gnomAD Exomes             PASS               NaN   \n",
      "23                 gnomAD Exomes             PASS               NaN   \n",
      "\n",
      "           Transcript HGVS Consequence  ... Homozygote Count Other  \\\n",
      "0   ENST00000302118.5         c.-64C>T  ...                      5   \n",
      "1   ENST00000302118.5         c.-59T>G  ...                      0   \n",
      "2   ENST00000302118.5         c.-54C>T  ...                      0   \n",
      "3   ENST00000302118.5         c.-54C>G  ...                      0   \n",
      "4   ENST00000302118.5         c.-49A>G  ...                      0   \n",
      "5   ENST00000302118.5         c.-48C>T  ...                      0   \n",
      "6   ENST00000302118.5         c.-47G>C  ...                      0   \n",
      "7   ENST00000302118.5         c.-45C>T  ...                      0   \n",
      "8   ENST00000302118.5         c.-42C>T  ...                      0   \n",
      "9   ENST00000302118.5         c.-34C>T  ...                      0   \n",
      "10  ENST00000302118.5         c.-29C>T  ...                      0   \n",
      "11  ENST00000302118.5         c.-28C>G  ...                      0   \n",
      "12  ENST00000302118.5         c.-26G>A  ...                      0   \n",
      "13  ENST00000302118.5         c.-23C>T  ...                      0   \n",
      "14  ENST00000302118.5         c.-22A>C  ...                      0   \n",
      "15  ENST00000302118.5         c.-12C>A  ...                      0   \n",
      "16  ENST00000302118.5         c.-11C>T  ...                      0   \n",
      "17  ENST00000302118.5         c.-10C>T  ...                      0   \n",
      "18  ENST00000302118.5          c.-8T>C  ...                      0   \n",
      "19  ENST00000302118.5          c.-4C>A  ...                      0   \n",
      "20  ENST00000302118.5          p.Met1?  ...                      0   \n",
      "21  ENST00000302118.5        p.Gly2Asp  ...                      0   \n",
      "22  ENST00000302118.5        p.Thr3Thr  ...                      0   \n",
      "23  ENST00000302118.5        p.Val4Ile  ...                      0   \n",
      "\n",
      "   Hemizygote Count Other Allele Count South Asian Allele Number South Asian  \\\n",
      "0                       0                        0                         0   \n",
      "1                       0                        0                         0   \n",
      "2                       0                        0                         0   \n",
      "3                       0                        0                         0   \n",
      "4                       0                        0                     24070   \n",
      "5                       0                        0                     24086   \n",
      "6                       0                        0                     24098   \n",
      "7                       0                        0                         0   \n",
      "8                       0                        0                         0   \n",
      "9                       0                        0                     24258   \n",
      "10                      0                        0                     24308   \n",
      "11                      0                        0                         0   \n",
      "12                      0                        0                     24292   \n",
      "13                      0                        0                     24244   \n",
      "14                      0                        0                     24250   \n",
      "15                      0                        0                     24102   \n",
      "16                      0                        0                     24064   \n",
      "17                      0                        0                     24062   \n",
      "18                      0                        0                     24002   \n",
      "19                      0                        0                     24022   \n",
      "20                      0                        1                     23972   \n",
      "21                      0                        1                     23906   \n",
      "22                      0                        0                     23922   \n",
      "23                      0                        0                     23864   \n",
      "\n",
      "    Homozygote Count South Asian  Hemizygote Count South Asian  chromosome  \\\n",
      "0                              0                             0        chr1   \n",
      "1                              0                             0        chr1   \n",
      "2                              0                             0        chr1   \n",
      "3                              0                             0        chr1   \n",
      "4                              0                             0        chr1   \n",
      "5                              0                             0        chr1   \n",
      "6                              0                             0        chr1   \n",
      "7                              0                             0        chr1   \n",
      "8                              0                             0        chr1   \n",
      "9                              0                             0        chr1   \n",
      "10                             0                             0        chr1   \n",
      "11                             0                             0        chr1   \n",
      "12                             0                             0        chr1   \n",
      "13                             0                             0        chr1   \n",
      "14                             0                             0        chr1   \n",
      "15                             0                             0        chr1   \n",
      "16                             0                             0        chr1   \n",
      "17                             0                             0        chr1   \n",
      "18                             0                             0        chr1   \n",
      "19                             0                             0        chr1   \n",
      "20                             0                             0        chr1   \n",
      "21                             0                             0        chr1   \n",
      "22                             0                             0        chr1   \n",
      "23                             0                             0        chr1   \n",
      "\n",
      "       start       end    value  \n",
      "0   55505385  55505535  202.571  \n",
      "1   55505385  55505535  202.571  \n",
      "2   55505385  55505535  202.571  \n",
      "3   55505385  55505535  202.571  \n",
      "4   55505385  55505535  202.571  \n",
      "5   55505385  55505535  202.571  \n",
      "6   55505385  55505535  202.571  \n",
      "7   55505385  55505535  202.571  \n",
      "8   55505385  55505535  202.571  \n",
      "9   55505385  55505535  202.571  \n",
      "10  55505385  55505535  202.571  \n",
      "11  55505385  55505535  202.571  \n",
      "12  55505385  55505535  202.571  \n",
      "13  55505385  55505535  202.571  \n",
      "14  55505385  55505535  202.571  \n",
      "15  55505385  55505535  202.571  \n",
      "16  55505385  55505535  202.571  \n",
      "17  55505385  55505535  202.571  \n",
      "18  55505385  55505535  202.571  \n",
      "19  55505385  55505535  202.571  \n",
      "20  55505385  55505535  202.571  \n",
      "21  55505385  55505535  202.571  \n",
      "22  55505385  55505535  202.571  \n",
      "23  55505385  55505535  202.571  \n",
      "\n",
      "[24 rows x 57 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Sample data\n",
    "data1 = gf\n",
    "df1 = pd.DataFrame(data1)\n",
    "\n",
    "data2 = rf\n",
    "df2 = pd.DataFrame(data2)\n",
    "\n",
    "\n",
    "# List to store merged rows\n",
    "merged_rows = []\n",
    "\n",
    "# Loop through each row in df1 and df2\n",
    "for index1, row1 in df1.iterrows():\n",
    "    for index2, row2 in df2.iterrows():\n",
    "        if row2['start'] <= row1['Position'] <= row2['end']:\n",
    "            # Combine the rows\n",
    "            combined_row = row1.to_dict()\n",
    "            combined_row.update(row2.to_dict())\n",
    "            merged_rows.append(combined_row)\n",
    "\n",
    "# Convert merged rows to DataFrame\n",
    "merged_df = pd.DataFrame(merged_rows)\n",
    "\n",
    "merged_df[['Position', 'start', 'end']]\n",
    "\n",
    "print(merged_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Position</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>55505447</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>55505452</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>55505457</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>55505457</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>55505462</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>55505463</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>55505464</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>55505466</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>55505469</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>55505477</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>55505482</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>55505483</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>55505485</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>55505488</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>55505489</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>55505499</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>55505500</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>55505501</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>55505503</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>55505507</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>55505512</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>55505515</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>55505519</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>55505520</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Position     start       end\n",
       "0   55505447  55505385  55505535\n",
       "1   55505452  55505385  55505535\n",
       "2   55505457  55505385  55505535\n",
       "3   55505457  55505385  55505535\n",
       "4   55505462  55505385  55505535\n",
       "5   55505463  55505385  55505535\n",
       "6   55505464  55505385  55505535\n",
       "7   55505466  55505385  55505535\n",
       "8   55505469  55505385  55505535\n",
       "9   55505477  55505385  55505535\n",
       "10  55505482  55505385  55505535\n",
       "11  55505483  55505385  55505535\n",
       "12  55505485  55505385  55505535\n",
       "13  55505488  55505385  55505535\n",
       "14  55505489  55505385  55505535\n",
       "15  55505499  55505385  55505535\n",
       "16  55505500  55505385  55505535\n",
       "17  55505501  55505385  55505535\n",
       "18  55505503  55505385  55505535\n",
       "19  55505507  55505385  55505535\n",
       "20  55505512  55505385  55505535\n",
       "21  55505515  55505385  55505535\n",
       "22  55505519  55505385  55505535\n",
       "23  55505520  55505385  55505535"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-76-a5b5cd7a4af5>:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  df2[df2.apply(lambda x: (df1['Position'] >= x['start']) & (df1['Position'] <= x['end']), axis=1).any()]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55505100</td>\n",
       "      <td>55505250</td>\n",
       "      <td>120.199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "      <td>202.571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55504905</td>\n",
       "      <td>55505055</td>\n",
       "      <td>103.534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55586360</td>\n",
       "      <td>55586510</td>\n",
       "      <td>79.1868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55411220</td>\n",
       "      <td>55411370</td>\n",
       "      <td>122.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55525845</td>\n",
       "      <td>55525995</td>\n",
       "      <td>73.5834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55534780</td>\n",
       "      <td>55534930</td>\n",
       "      <td>104.872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55509965</td>\n",
       "      <td>55510115</td>\n",
       "      <td>134.802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55506205</td>\n",
       "      <td>55506355</td>\n",
       "      <td>98.7568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55504305</td>\n",
       "      <td>55504455</td>\n",
       "      <td>112.666</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome     start       end    value\n",
       "0       chr1  55505100  55505250  120.199\n",
       "1       chr1  55505385  55505535  202.571\n",
       "2       chr1  55504905  55505055  103.534\n",
       "3       chr1  55586360  55586510  79.1868\n",
       "4       chr1  55411220  55411370   122.74\n",
       "5       chr1  55525845  55525995  73.5834\n",
       "6       chr1  55534780  55534930  104.872\n",
       "7       chr1  55509965  55510115  134.802\n",
       "8       chr1  55506205  55506355  98.7568\n",
       "9       chr1  55504305  55504455  112.666"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2[df2.apply(lambda x: (df1['Position'] >= x['start']) & (df1['Position'] <= x['end']), axis=1).any()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55505100</td>\n",
       "      <td>55505250</td>\n",
       "      <td>120.199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55505385</td>\n",
       "      <td>55505535</td>\n",
       "      <td>202.571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55504905</td>\n",
       "      <td>55505055</td>\n",
       "      <td>103.534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55586360</td>\n",
       "      <td>55586510</td>\n",
       "      <td>79.1868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55411220</td>\n",
       "      <td>55411370</td>\n",
       "      <td>122.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55525845</td>\n",
       "      <td>55525995</td>\n",
       "      <td>73.5834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55534780</td>\n",
       "      <td>55534930</td>\n",
       "      <td>104.872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55509965</td>\n",
       "      <td>55510115</td>\n",
       "      <td>134.802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55506205</td>\n",
       "      <td>55506355</td>\n",
       "      <td>98.7568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>55504305</td>\n",
       "      <td>55504455</td>\n",
       "      <td>112.666</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome     start       end    value\n",
       "0       chr1  55505100  55505250  120.199\n",
       "1       chr1  55505385  55505535  202.571\n",
       "2       chr1  55504905  55505055  103.534\n",
       "3       chr1  55586360  55586510  79.1868\n",
       "4       chr1  55411220  55411370   122.74\n",
       "5       chr1  55525845  55525995  73.5834\n",
       "6       chr1  55534780  55534930  104.872\n",
       "7       chr1  55509965  55510115  134.802\n",
       "8       chr1  55506205  55506355  98.7568\n",
       "9       chr1  55504305  55504455  112.666"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>rsIDs</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternate</th>\n",
       "      <th>Source</th>\n",
       "      <th>Filters - exomes</th>\n",
       "      <th>Filters - genomes</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>HGVS Consequence</th>\n",
       "      <th>...</th>\n",
       "      <th>Homozygote Count European (non-Finnish)</th>\n",
       "      <th>Hemizygote Count European (non-Finnish)</th>\n",
       "      <th>Allele Count Other</th>\n",
       "      <th>Allele Number Other</th>\n",
       "      <th>Homozygote Count Other</th>\n",
       "      <th>Hemizygote Count Other</th>\n",
       "      <th>Allele Count South Asian</th>\n",
       "      <th>Allele Number South Asian</th>\n",
       "      <th>Homozygote Count South Asian</th>\n",
       "      <th>Hemizygote Count South Asian</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>55505447</td>\n",
       "      <td>rs45448095</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>gnomAD Genomes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ENST00000302118.5</td>\n",
       "      <td>c.-64C&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>107</td>\n",
       "      <td>0</td>\n",
       "      <td>108</td>\n",
       "      <td>1086</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>55505452</td>\n",
       "      <td>rs576575389</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>gnomAD Genomes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ENST00000302118.5</td>\n",
       "      <td>c.-59T&gt;G</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1086</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>55505457</td>\n",
       "      <td>rs545454601</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>gnomAD Genomes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ENST00000302118.5</td>\n",
       "      <td>c.-54C&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1088</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>55505457</td>\n",
       "      <td>rs545454601</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>gnomAD Genomes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ENST00000302118.5</td>\n",
       "      <td>c.-54C&gt;G</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1088</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>55505462</td>\n",
       "      <td>rs1043549074</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000302118.5</td>\n",
       "      <td>c.-49A&gt;G</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4680</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24070</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1192</th>\n",
       "      <td>1</td>\n",
       "      <td>55529286</td>\n",
       "      <td>rs771949878</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000302118.5</td>\n",
       "      <td>c.*29G&gt;A</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5670</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>29146</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1193</th>\n",
       "      <td>1</td>\n",
       "      <td>55529301</td>\n",
       "      <td>rs1340440495</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000302118.5</td>\n",
       "      <td>c.*44G&gt;A</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5150</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>26710</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1194</th>\n",
       "      <td>1</td>\n",
       "      <td>55529306</td>\n",
       "      <td>rs773075461</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000302118.5</td>\n",
       "      <td>c.*49G&gt;T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4988</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>25776</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1195</th>\n",
       "      <td>1</td>\n",
       "      <td>55529307</td>\n",
       "      <td>rs760320200</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>gnomAD Exomes</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000302118.5</td>\n",
       "      <td>c.*50A&gt;G</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>25678</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1196</th>\n",
       "      <td>1</td>\n",
       "      <td>55529324</td>\n",
       "      <td>rs764673359</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>gnomAD Genomes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ENST00000302118.5</td>\n",
       "      <td>c.*67G&gt;A</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1086</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1197 rows Ã— 53 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chromosome  Position         rsIDs Reference Alternate          Source  \\\n",
       "0              1  55505447    rs45448095         C         T  gnomAD Genomes   \n",
       "1              1  55505452   rs576575389         T         G  gnomAD Genomes   \n",
       "2              1  55505457   rs545454601         C         T  gnomAD Genomes   \n",
       "3              1  55505457   rs545454601         C         G  gnomAD Genomes   \n",
       "4              1  55505462  rs1043549074         A         G   gnomAD Exomes   \n",
       "...          ...       ...           ...       ...       ...             ...   \n",
       "1192           1  55529286   rs771949878         G         A   gnomAD Exomes   \n",
       "1193           1  55529301  rs1340440495         G         A   gnomAD Exomes   \n",
       "1194           1  55529306   rs773075461         G         T   gnomAD Exomes   \n",
       "1195           1  55529307   rs760320200         A         G   gnomAD Exomes   \n",
       "1196           1  55529324   rs764673359         G         A  gnomAD Genomes   \n",
       "\n",
       "     Filters - exomes Filters - genomes         Transcript HGVS Consequence  \\\n",
       "0                 NaN              PASS  ENST00000302118.5         c.-64C>T   \n",
       "1                 NaN              PASS  ENST00000302118.5         c.-59T>G   \n",
       "2                 NaN              PASS  ENST00000302118.5         c.-54C>T   \n",
       "3                 NaN              PASS  ENST00000302118.5         c.-54C>G   \n",
       "4                PASS               NaN  ENST00000302118.5         c.-49A>G   \n",
       "...               ...               ...                ...              ...   \n",
       "1192             PASS               NaN  ENST00000302118.5         c.*29G>A   \n",
       "1193             PASS               NaN  ENST00000302118.5         c.*44G>A   \n",
       "1194             PASS               NaN  ENST00000302118.5         c.*49G>T   \n",
       "1195             PASS               NaN  ENST00000302118.5         c.*50A>G   \n",
       "1196              NaN              PASS  ENST00000302118.5         c.*67G>A   \n",
       "\n",
       "      ... Homozygote Count European (non-Finnish)  \\\n",
       "0     ...                                     107   \n",
       "1     ...                                       0   \n",
       "2     ...                                       0   \n",
       "3     ...                                       0   \n",
       "4     ...                                       0   \n",
       "...   ...                                     ...   \n",
       "1192  ...                                       0   \n",
       "1193  ...                                       0   \n",
       "1194  ...                                       0   \n",
       "1195  ...                                       0   \n",
       "1196  ...                                       0   \n",
       "\n",
       "     Hemizygote Count European (non-Finnish) Allele Count Other  \\\n",
       "0                                          0                108   \n",
       "1                                          0                  0   \n",
       "2                                          0                  0   \n",
       "3                                          0                  0   \n",
       "4                                          0                  0   \n",
       "...                                      ...                ...   \n",
       "1192                                       0                  0   \n",
       "1193                                       0                  0   \n",
       "1194                                       0                  0   \n",
       "1195                                       0                  0   \n",
       "1196                                       0                  0   \n",
       "\n",
       "     Allele Number Other  Homozygote Count Other Hemizygote Count Other  \\\n",
       "0                   1086                       5                      0   \n",
       "1                   1086                       0                      0   \n",
       "2                   1088                       0                      0   \n",
       "3                   1088                       0                      0   \n",
       "4                   4680                       0                      0   \n",
       "...                  ...                     ...                    ...   \n",
       "1192                5670                       0                      0   \n",
       "1193                5150                       0                      0   \n",
       "1194                4988                       0                      0   \n",
       "1195                5000                       0                      0   \n",
       "1196                1086                       0                      0   \n",
       "\n",
       "      Allele Count South Asian  Allele Number South Asian  \\\n",
       "0                            0                          0   \n",
       "1                            0                          0   \n",
       "2                            0                          0   \n",
       "3                            0                          0   \n",
       "4                            0                      24070   \n",
       "...                        ...                        ...   \n",
       "1192                         0                      29146   \n",
       "1193                         0                      26710   \n",
       "1194                         0                      25776   \n",
       "1195                         0                      25678   \n",
       "1196                         0                          0   \n",
       "\n",
       "      Homozygote Count South Asian  Hemizygote Count South Asian  \n",
       "0                                0                             0  \n",
       "1                                0                             0  \n",
       "2                                0                             0  \n",
       "3                                0                             0  \n",
       "4                                0                             0  \n",
       "...                            ...                           ...  \n",
       "1192                             0                             0  \n",
       "1193                             0                             0  \n",
       "1194                             0                             0  \n",
       "1195                             0                             0  \n",
       "1196                             0                             0  \n",
       "\n",
       "[1197 rows x 53 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
